---
title: "16s_roots"
author: "Beant Kapoor"
date: "3/7/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(echo=FALSE, warning=FALSE, message=FALSE)
```

### General Comment - Objects in this file contain the name 'otu' however they represent ASV in this case.

```{r Input Data and subset}
#We already have imported the otu file and taxonomy file from DADA2 to this R project/environment. We just need to import the metadata for roots niche and subset the data accordingly.
roots_metadata <- read.table("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/metadata/16s_roots.txt", header = TRUE, fill = TRUE, strip.white = TRUE)
roots_sample_names <- roots_metadata$Sample

#Here I use the subset function to pull out just the samples belonging to roots
roots_otu <- subset(whole_otu, rownames(whole_otu) %in% roots_sample_names)
```

```{r Data Clean-up}
#Let's remove the columns which are not needed
roots_otu_only <- roots_otu[, 3:170189]

#Here I am removing any OTU that is not present in any samples. What the below function is doing is summing the columns (OTUs) and any OTU that equals 0 is removed from the dataset. This will hopefully speed up downstream analyses.
roots_pure <- roots_otu_only[, colSums(roots_otu_only) > 0]
```

```{r Rarefaction}
#Below I create the rarefaction curves for each sample using the rarecurve function from vegan. In this case we should use the object 'roots_pure'
library(vegan)
rare_curve_roots <- rarecurve(roots_pure)

#I then create an object that spans the range of the ESV richness. This will be used to plot the vertical line in my ggplot rarefaction curve. 
f_roots<-(0:5000)

#I then create objects containing the x and y coordinates values for each stratum.
sequences_roots40<-attr(rare_curve_roots[[40]],which="Subsample")
sequences_roots39<-attr(rare_curve_roots[[39]],which="Subsample")
sequences_roots38<-attr(rare_curve_roots[[38]],which="Subsample")
sequences_roots37<-attr(rare_curve_roots[[37]],which="Subsample")
sequences_roots36<-attr(rare_curve_roots[[36]],which="Subsample")
sequences_roots35<-attr(rare_curve_roots[[35]],which="Subsample")
sequences_roots34<-attr(rare_curve_roots[[34]],which="Subsample")
sequences_roots33<-attr(rare_curve_roots[[33]],which="Subsample")
sequences_roots32<-attr(rare_curve_roots[[32]],which="Subsample")
sequences_roots31<-attr(rare_curve_roots[[31]],which="Subsample")
sequences_roots30<-attr(rare_curve_roots[[30]],which="Subsample")
sequences_roots29<-attr(rare_curve_roots[[29]],which="Subsample")
sequences_roots28<-attr(rare_curve_roots[[28]],which="Subsample")
sequences_roots27<-attr(rare_curve_roots[[27]],which="Subsample")
sequences_roots26<-attr(rare_curve_roots[[26]],which="Subsample")
sequences_roots25<-attr(rare_curve_roots[[25]],which="Subsample")
sequences_roots24<-attr(rare_curve_roots[[24]],which="Subsample")
sequences_roots23<-attr(rare_curve_roots[[23]],which="Subsample")
sequences_roots22<-attr(rare_curve_roots[[22]],which="Subsample")
sequences_roots21<-attr(rare_curve_roots[[21]],which="Subsample")
sequences_roots20<-attr(rare_curve_roots[[20]],which="Subsample")
sequences_roots19<-attr(rare_curve_roots[[19]],which="Subsample")
sequences_roots18<-attr(rare_curve_roots[[18]],which="Subsample")
sequences_roots17<-attr(rare_curve_roots[[17]],which="Subsample")
sequences_roots16<-attr(rare_curve_roots[[16]],which="Subsample")
sequences_roots15<-attr(rare_curve_roots[[15]],which="Subsample")
sequences_roots14<-attr(rare_curve_roots[[14]],which="Subsample")
sequences_roots13<-attr(rare_curve_roots[[13]],which="Subsample")
sequences_roots12<-attr(rare_curve_roots[[12]],which="Subsample")
sequences_roots11<-attr(rare_curve_roots[[11]],which="Subsample")
sequences_roots10<-attr(rare_curve_roots[[10]],which="Subsample")
sequences_roots9<-attr(rare_curve_roots[[9]],which="Subsample")
sequences_roots8<-attr(rare_curve_roots[[8]],which="Subsample")
sequences_roots7<-attr(rare_curve_roots[[7]],which="Subsample")
sequences_roots6<-attr(rare_curve_roots[[6]],which="Subsample")
sequences_roots5<-attr(rare_curve_roots[[5]],which="Subsample")
sequences_roots4<-attr(rare_curve_roots[[4]],which="Subsample")
sequences_roots3<-attr(rare_curve_roots[[3]],which="Subsample")
sequences_roots2<-attr(rare_curve_roots[[2]],which="Subsample")
sequences_roots1<-attr(rare_curve_roots[[1]],which="Subsample")

#Here I plot the rarefaction curves. 

library(ggplot2)
library(ggpubr)

rarefaction_chart_roots <- ggplot() +
  geom_line(aes(x=sequences_roots40,y=rare_curve_roots[[40]]),size=1)+
  geom_line(aes(x=sequences_roots39,y=rare_curve_roots[[39]]), size=1)+
  geom_line(aes(x=sequences_roots38,y=rare_curve_roots[[38]]),size=1)+
  geom_line(aes(x=sequences_roots37,y=rare_curve_roots[[37]]),size=1)+
  geom_line(aes(x=sequences_roots36,y=rare_curve_roots[[36]]),size=1)+
  geom_line(aes(x=sequences_roots35,y=rare_curve_roots[[35]]),size=1)+
  geom_line(aes(x=sequences_roots34,y=rare_curve_roots[[34]]), size=1)+
  geom_line(aes(x=sequences_roots33,y=rare_curve_roots[[33]]), size=1)+
  geom_line(aes(x=sequences_roots32,y=rare_curve_roots[[32]]),size=1)+
  geom_line(aes(x=sequences_roots31,y=rare_curve_roots[[31]]),size=1 )+
  geom_line(aes(x=sequences_roots30,y=rare_curve_roots[[30]]),size=1)+
  geom_line(aes(x=sequences_roots29,y=rare_curve_roots[[29]]),size=1)+
  geom_line(aes(x=sequences_roots28,y=rare_curve_roots[[28]]),size=1)+
  geom_line(aes(x=sequences_roots27,y=rare_curve_roots[[27]]),size=1)+
  geom_line(aes(x=sequences_roots26,y=rare_curve_roots[[26]]),size=1)+
  geom_line(aes(x=sequences_roots25,y=rare_curve_roots[[25]]), size=1)+
  geom_line(aes(x=sequences_roots24,y=rare_curve_roots[[24]]),size=1)+
  geom_line(aes(x=sequences_roots23,y=rare_curve_roots[[23]]),size=1)+
  geom_line(aes(x=sequences_roots22,y=rare_curve_roots[[22]]),size=1)+
  geom_line(aes(x=sequences_roots21,y=rare_curve_roots[[21]]),size=1)+
  geom_line(aes(x=sequences_roots20,y=rare_curve_roots[[20]]),size=1)+
  geom_line(aes(x=sequences_roots19,y=rare_curve_roots[[19]]), size=1)+
  geom_line(aes(x=sequences_roots18,y=rare_curve_roots[[18]]),size=1)+
  geom_line(aes(x=sequences_roots17,y=rare_curve_roots[[17]]),size=1)+
  geom_line(aes(x=sequences_roots16,y=rare_curve_roots[[16]]),size=1)+
  geom_line(aes(x=sequences_roots15,y=rare_curve_roots[[15]]),size=1)+
  geom_line(aes(x=sequences_roots14,y=rare_curve_roots[[14]]), size=1)+
  geom_line(aes(x=sequences_roots13,y=rare_curve_roots[[13]]), size=1)+
  geom_line(aes(x=sequences_roots12,y=rare_curve_roots[[12]]),size=1)+
  geom_line(aes(x=sequences_roots11,y=rare_curve_roots[[11]]),size=1 )+
  geom_line(aes(x=sequences_roots10,y=rare_curve_roots[[10]]),size=1)+
  geom_line(aes(x=sequences_roots9,y=rare_curve_roots[[9]]),size=1)+
  geom_line(aes(x=sequences_roots8,y=rare_curve_roots[[8]]),size=1)+
  geom_line(aes(x=sequences_roots7,y=rare_curve_roots[[7]]),size=1)+
  geom_line(aes(x=sequences_roots6,y=rare_curve_roots[[6]]),size=1)+
  geom_line(aes(x=sequences_roots5,y=rare_curve_roots[[5]]), size=1)+
  geom_line(aes(x=sequences_roots4,y=rare_curve_roots[[4]]),size=1)+
  geom_line(aes(x=sequences_roots3,y=rare_curve_roots[[3]]),size=1)+
  geom_line(aes(x=sequences_roots2,y=rare_curve_roots[[2]]),size=1)+
  geom_line(aes(x=sequences_roots1,y=rare_curve_roots[[1]]),size=1)+
  geom_vline(xintercept = 3614, linetype = "dashed")+
  ylim(0, 5000)+
  xlim(0, 50000)+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
       panel.border=element_rect(color="black", size=1, fill=NA))+
 theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+ 
labs(x=("Sequencing Depth"), y=expression(bold(paste("OTU Richness", sep=""))))+
  labs(title = 'Roots 16S')
rarefaction_chart_roots

ggsave(filename = 'roots_16s_rarefaction.png', path = "/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Roots_16s/Results", rarefaction_chart_roots, dpi = 300, width = 5, height = 5, units = 'in')

#Here we are rarefying the data using the rrarefy function from vegan. I first use the sort function to determine the lowest number of sequences in a sample. 
sort(rowSums(roots_pure)) #

#Then using the rrarefy function from vegan, I rarefy my samples. The rarefaction cutoff or "sample" value is determined by evaluating my rarefaction curves and choosing the number of sequences that minimizes sample loss but maximizes sampling depth.
rare_otu_roots <- as.data.frame(rrarefy(roots_pure, sample = 3614)) #In this case we are not losing any samples therefore we do not need to perform any subsetting.

#I am now removing OTUs present in the OTU table that were only present in samples removed following rarefaction. These OTUs will have column sums of 0.
rare_otu_roots <- rare_otu_roots[, colSums(rare_otu_roots) > 0]

#The following line is for saving the rarefied OTU table. This file is later reloaded to be used in downstream analyses. This is done to maintain consistency of results because rarefying generates slightly different results each time.
#write.table(rare_otu_roots, "/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Roots_16s/Tables_from_R/16s_roots_rarefied_otu.shared", sep = '\t', row.names = TRUE, col.names = TRUE)

#I am now removing singleton OTUs (OTUs with only 1 representative sequence following rarefaction). This is to limit the effects of rare species on our distance matrices used during ordination.  
rare_otu_roots_no_singletons <- rare_otu_roots[, colSums(rare_otu_roots) > 1]

#The following line is for saving the rarefied OTU table with singletons removed. This file is later reloaded to be used in downstream analyses. This is done to maintain consistency of results because rarefying generates slightly different results each time.
#write.table(rare_otu_roots_no_singletons, "/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Roots_16s/Tables_from_R/16s_roots_rarefied_no_singletons.shared", sep = '\t', row.names = TRUE, col.names = TRUE)
```

```{r Relative abundance chart - Phylum}
#Now that samples have been rarefied I can perform alpha and beta diversity analyses on my data sets. To begin I first reformat my OTU table so that I can do some filtering and add taxonomy information.
rare_otu_roots_no_singletons <- read.table("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Roots_16s/Tables_from_R/16s_roots_rarefied_no_singletons.shared", header = TRUE, sep = '\t')

rare_otu_roots <- read.table("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Roots_16s/Tables_from_R/16s_roots_rarefied_otu.shared", header = TRUE, sep = "\t")

#I now transpose my OTU table so that I can begin to add in taxonomy information.
trans_otu_roots <- as.data.frame(t(rare_otu_roots_no_singletons))

#The below function is asking specifically for the OTU IDs in the roots OTU table. These are the row names. I will use these to subset the taxonomy file.
roots_otu_names <- rownames(trans_otu_roots)
roots_taxonomy <- subset(taxonomy, rownames(taxonomy) %in% roots_otu_names)

#Now I am subsetting the new taxonomy file to include only the taxonomy information stored in column 2. 
roots_taxonomy_only <- roots_taxonomy[, 2]

#Below I am using the separate function from tidyr to split the taxonomy strings into columns by semi-colon so that I can rename the OTUs to give them more meaning for downstream analyses.
library(tidyr)

#The SILVA database only provides classifications up to genus level, new columns are created only up to Genus level.
taxonomy_labels<-c("kingdom","phylum","class","order","family","genus")

#Below I create a new data frame with a column containing taxonomy information.
roots_otu_taxonomy <- data.frame(taxonomy = roots_taxonomy_only, trans_otu_roots)

#I then separate the taxonomy strings into new columns labeled with the taxonomy labels saved in the taxonomy_labels object. Strings are being separated based on the presence of semi-colons.
roots_otu_taxonomy <- separate(roots_otu_taxonomy, col = taxonomy, into = taxonomy_labels, sep = ';')

#Below I create a data frame that consists of phylum assignment and the rarefied OTU table.
roots_otu_phylum <- data.frame(phylum = roots_otu_taxonomy$phylum, trans_otu_roots)

#I then remove the assignment confidence values contained in parentheses found in each taxonomic string using the sub function.
roots_otu_phylum$phylum <- sub("\\(.*)","",x = roots_otu_phylum$phylum)

#The following code is for the generation of phylum relative abundance stacked bar charts. I first sum each OTU by the phylum it belongs to. 
library(plyr)
roots_summed_otu_phylum <- as.data.frame(ddply(roots_otu_phylum, .(phylum), colwise(sum)))
rownames(roots_summed_otu_phylum) <- make.names(roots_summed_otu_phylum$phylum)
roots_summed_otu_phylum <- roots_summed_otu_phylum[, 2:41]

#I then transpose the data frame so that sample names are now row names and column names are OTU names.
trans_roots_otu_summed_phylum <- as.data.frame(t(roots_summed_otu_phylum))
rownames(trans_roots_otu_summed_phylum) <- sub('X', '', x = rownames(trans_roots_otu_summed_phylum))
```

```{r Continuing relative abundace chart at Phylum level}
roots_metadata_for_rel_abund <- read_excel("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/metadata/16s_metadata_without_plot.xlsx", sheet = 2)
trans_roots_phylum_metadata <- merge(roots_metadata_for_rel_abund, trans_roots_otu_summed_phylum, by.x = 1, by.y = 0)

#Then we can delete the sample name column as we don't need that.
trans_roots_phylum_metadata <- trans_roots_phylum_metadata[, 2:24]

#I then sum each OTU by Treatment column and subset to exclude sample names. This is because the following functions only work on matrices containing numeric data.
roots_phylum_summed_treamtment <- as.data.frame(ddply(trans_roots_phylum_metadata, .(Treatment), colwise(sum)))
roots_phylum <- roots_phylum_summed_treamtment[, 2:23]

#I am now creating an other category. Other includes those OTUs belonging to a phylum that comprises less than 1% of the total community. This new object is reffered to as phylum_others.
roots_phylum_others <- roots_phylum[, colSums(roots_phylum)/sum(roots_phylum) <= 0.01]
roots_phylum_others_sum <- rowSums(roots_phylum_others)
roots_phylum_regular <- roots_phylum[, colSums(roots_phylum)/sum(roots_phylum) > 0.01]

#I then create a dataframe containing the treatment information, the Other column and the remaining phyla
roots_phylum_reg_others <- data.frame(Treatment = roots_phylum_summed_treamtment$Treatment, roots_phylum_regular, Others = roots_phylum_others_sum)

#These values are then converted to relative abundance using the decostand function from vegan.
library(vegan)
roots_phylum_rel_abund <- decostand(roots_phylum_reg_others[, 2:11], method = 'total')
roots_phylum_rel_abund <- data.frame(Treatment = roots_phylum_reg_others$Treatment, roots_phylum_rel_abund)

#Below I use the melt function from the data.table package to reformat the data into stacked bar graph format. 
library(data.table)
library(ggplot2)
library(ggpubr)
library(RColorBrewer)
roots_phylum_melt <- melt(roots_phylum_rel_abund, id.vars = 'Treatment', variable.name = 'Phylum')
roots_colors_n_phylum <- length(unique(roots_phylum_melt$Phylum))
roots_phylum_melt$Treatment <- as.character(roots_phylum_melt$Treatment)
roots_phylum_melt$Treatment <- factor(roots_phylum_melt$Treatment, levels = c("Control 2018", "Control 2019", "Pre-burned 2018", "Post-burned 2019"))

tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Roots_16s/Results/16s_roots_relative_abundance_phylum.tiff", units = 'in', height = 7, width = 7, res = 200)
roots_rel_chart_phylum <- ggplot(roots_phylum_melt, aes(x = Treatment, y = value, fill = Phylum))+
  geom_bar(stat = 'identity', show.legend = TRUE, color = 'black')+
  scale_fill_manual(values = colorRampPalette(brewer.pal(7, 'Accent'))(roots_colors_n_phylum))+
  xlab('Roots')+
  ylab('Relative Abundance')+
  ggtitle('Roots - Phylum level')+
  theme(panel.border = element_blank(),
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.title.x = element_blank(), 
        axis.text.x = element_text(size = 12, color = 'black', angle = 90), 
        axis.ticks.x = element_blank(), 
        axis.text.y = element_text(size = 14), 
        axis.line.y.left = element_line(), 
        axis.title.y = element_blank(),
        legend.title = element_text(face = "bold", size = 14),
        legend.text = element_text(size = 14))
roots_rel_chart_phylum
```

```{r}
roots_phylum_percent <- roots_phylum_rel_abund * 100 
roots_phylum_percent <- roots_phylum_percent[, -1]
roots_phylum_percent <- data.frame(Treatment = roots_phylum_rel_abund$Treatment, roots_phylum_percent)
is.num <- sapply(roots_phylum_percent, is.numeric)
roots_phylum_percent[is.num] <- lapply(roots_phylum_percent[is.num], round, 2)
roots_proteobac_mean <- mean(roots_phylum_percent$Proteobacteria)
roots_proteobac_mean
```


```{r Relative abundance - Class}
roots_otu_class <- data.frame(class = roots_otu_taxonomy$class, trans_otu_roots)

#I then remove the assignment confidence values contained in parentheses found in each taxonomic string using the sub function.
roots_otu_class$class <- sub("\\(.*)", '', x = roots_otu_class$class)
roots_summed_class <- as.data.frame(ddply(roots_otu_class, .(class), colwise(sum)))
rownames(roots_summed_class) <- make.names(roots_summed_class$class)
roots_summed_class <- roots_summed_class[, 2:41]

#I then transpose the dataframe and merge the metadata information
trans_roots_class <- as.data.frame(t(roots_summed_class))
rownames(trans_roots_class) <- sub('X', '', x = rownames(trans_roots_class))
roots_class_metadata <- merge(roots_metadata_for_rel_abund, trans_roots_class, by.x = 1, by.y = 0)

#Then we can delete the column with the sample names as they are not needed and then sum by treatment
roots_class_metadata <- roots_class_metadata[, 2:51]
roots_class_summed_treamtent <- as.data.frame(ddply(roots_class_metadata, .(Treatment), colwise(sum)))

#I then subset the data so that only numerical data is present and creating a new class of objects, other, that includes OTUs from classes that represent 1% or less of the total community.
roots_class <- roots_class_summed_treamtent[, 2:50]
roots_class_others <- roots_class[, colSums(roots_class)/sum(roots_class) <= 0.01]
roots_class_others_sum <- rowSums(roots_class_others)
roots_class_regular <- roots_class[, colSums(roots_class)/sum(roots_class) > 0.01]

#I then create a new dataframe that contains all of the class data and determine relative abundace using decostand function from vegan
roots_class_reg_others <- data.frame(Treatment = roots_class_summed_treamtent$Treatment, roots_class_regular, Others = roots_class_others_sum)
roots_class_rel_abund <- decostand(roots_class_reg_others[, 2:12], method = 'total')
roots_class_rel_abund <- data.frame(Treatment = roots_class_reg_others$Treatment, roots_class_rel_abund)

#Below I use the melt function from the data.table package to reformat the data into stacked bar graph format.
library(data.table)
library(ggplot2)
library(ggpubr)
library(RColorBrewer)
roots_class_melt <- melt(roots_class_rel_abund, id.vars = 'Treatment', variable.name = 'Class')
roots_class_n_color <- length(unique(roots_class_melt$Class))
roots_class_melt$Treatment <- as.character(roots_class_melt$Treatment)
roots_class_melt$Treatment <- factor(roots_class_melt$Treatment, levels = c("Control 2018", "Control 2019", "Pre-burned 2018", "Post-burned 2019"))

tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/roots_16s/Results/16s_roots_relative_abundance_class.tiff", units = 'in', width = 7, height = 7, res = 200)
roots_rel_chart_class <- ggplot(roots_class_melt, aes(x = Treatment, y = value, fill = Class))+
  geom_bar(stat = 'identity', show.legend = TRUE, color = 'black')+
  scale_fill_manual(values = colorRampPalette(brewer.pal(7, 'Accent'))(roots_class_n_color))+
  xlab('Roots')+
  ylab('Relative Abundance')+
  ggtitle('Roots - Class level')+
  theme(panel.border = element_blank(), 
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.title.x = element_blank(), 
        axis.text.x = element_text(size = 12, color = 'black', angle = 90), 
        axis.ticks.x = element_blank(), 
        axis.text.y = element_text(size = 14), 
        axis.line.y.left = element_line(), 
        axis.title.y = element_blank())
dev.off()

#Let's join the relative abundance charts at class level
tiff(filename = "/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Roots_16s/Results/16s_roots_relative_abundance_phylum_class.tiff", units = 'in', width = 10, height = 7, res = 200)
ggarrange(roots_rel_chart_phylum, roots_rel_chart_class,  nrow = 1, labels = c("I", "II"), align = 'h')
dev.off()
```

# The purpose of the following chunk is to create Shannon Alpha diversity boxplots across treatments

```{r Alpha Diversity}
# We will calculate richness and create a dataframe including richness with metadata
# Calculate the species richness using the richness function of vegan. We are going to work with the object named 'rare_otu_roots' because the structure of this dataset behaves as vegan community matrix data.

#The following function from vegan will give us the shannon diversity index
roots_alpha_diversity <- diversity(rare_otu_roots, index = 'shannon')
roots_alpha_diversity <- as.data.frame(roots_alpha_diversity)

#Adding in the metadata for graph
roots_alpha_metadata <- merge(roots_metadata_for_rel_abund, roots_alpha_diversity, by.x = 1, by.y = 0)
rownames(roots_alpha_metadata) <- make.names(roots_alpha_metadata$Sample)
roots_alpha_metadata <- roots_alpha_metadata[, 2:3]
rownames(roots_alpha_metadata) <- sub('X', '', rownames(roots_alpha_metadata))
roots_alpha_metadata$Treatment <- as.character(roots_alpha_metadata$Treatment)
roots_alpha_metadata$Treatment <- factor(roots_alpha_metadata$Treatment, levels = c('Pre_control_A', 'Post_control_A', 'Pre_control_D', 'Post_control_D', 'Pre_burn_B', 'Post_burn_B', 'Pre_burn_C', 'Post_burn_C'))

#Let's make the graph now
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/roots_16s/Results/16s_roots_alpha_shannon_index.tiff", units = 'in', width = 10, height = 7, res = 200)
ggplot(roots_alpha_metadata, aes(x = Treatment, y = roots_alpha_diversity, fill = Treatment))+
  geom_boxplot(outlier.shape = NA)+
  ggtitle('Roots 16S Alpha Diversity')+
  ylab('Shannon Diversity Index')+
  xlab('Treatment')+
  theme_linedraw()+
  theme(axis.title.x = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(plot.title = element_text('Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.x = element_text(family = 'Times New Roman', size = 14, colour = 'black', face = 'bold', angle = 90))+
  theme(axis.title.y = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.y = element_text(family = 'Times New Roman', size = 14, face = 'bold'))+
  theme(legend.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(legend.text = element_text(size = 18, family = 'Times New Roman'))+
  scale_fill_brewer(palette = 'Set1')+
  guides(fill=guide_legend(title = 'Treatment'))
dev.off()

#Let's make the graph now without legend
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/roots_16s/Results/16s_roots_shannon_without_legend.tiff", units = 'in', width = 10, height = 7, res = 200)
roots_alpha_without_legend <- ggplot(roots_alpha_metadata, aes(x = Treatment, y = roots_alpha_diversity, fill = Treatment))+
  geom_boxplot(outlier.shape = NA)+
  ggtitle('Roots 16S Alpha Diversity')+
  ylab('Shannon Index')+
  xlab('Treatment')+
  theme_linedraw()+
  theme(axis.title.x = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(plot.title = element_text('Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.x = element_text(family = 'Times New Roman', size = 14, colour = 'black', face = 'bold', angle = 90))+
  theme(axis.title.y = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.y = element_text(family = 'Times New Roman', size = 14, face = 'bold'))+
  theme(legend.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(legend.text = element_text(size = 18, family = 'Times New Roman'))+
  theme(legend.position = 'none')+
  scale_fill_brewer(palette = 'Set1')+
  guides(fill=guide_legend(title = 'Treatment'))
dev.off()
```

```{r}
roots_class_percent <- roots_class_rel_abund * 100 
roots_class_percent <- roots_class_percent[, -1]
roots_class_percent <- data.frame(Treatment = roots_class_rel_abund$Treatment, roots_class_percent)
is.num <- sapply(roots_class_percent, is.numeric)
roots_class_percent[is.num] <- lapply(roots_class_percent[is.num], round, 2)
roots_actinobacteria_mean <- mean(roots_class_percent$Actinobacteria)
roots_actinobacteria_mean
```

# The following code is used to generate the Principal Coordinate Analysis chart

```{r}
#We will start with the object named 'rare_otu_roots_no_singletons'
library(vegan)
rare_roots_rel_abund_for_pcoa <- decostand(rare_otu_roots_no_singletons, method = 'total')

#Need to load the library 'ape' and use the vegdist function to create a dissimilarity matrix using the Bray Curtis distance
library(ape)
roots_dist <- vegdist(rare_roots_rel_abund_for_pcoa, method = 'bray')
roots_pcoa <- pcoa(roots_dist)
roots_pcoa_vectors <- as.data.frame(roots_pcoa$vectors)
roots_pcoa_vectors_metadata <- merge(roots_metadata_for_rel_abund, roots_pcoa_vectors, by.x = 1, by.y = 0)
biplot(roots_pcoa, plot.axes = c(1, 2))
roots_pcoa_scores <- data.frame(PC1 = roots_pcoa_vectors$Axis.1, PC2 = roots_pcoa_vectors$Axis.2)
roots_pcoa_ellipse <- ordiellipse(roots_pcoa_scores, roots_pcoa_vectors_metadata$Treatment)

#Plot the eigenvalues and interpret
barplot(roots_pcoa$values$Relative_eig[1:10])

# Find out the percentage variance explained by the first two principal axes
roots_pcoa$values$Relative_eig[1] * 100 #16.36
roots_pcoa$values$Relative_eig[2] * 100 #9.68

#Extract the plot scores from first two PCoA axes
roots_pcoa_axes <- roots_pcoa$vectors[, c(1, 2)]

#In order to change the color scheme of PCoA we will have to add one more column to the dataframe 'roots_pcoa_vectors_metadata'
roots_treatment_pcoa_color <- c('Burn_B', 'Burn_B','Burn_B','Burn_B','Burn_B', 'Burn_C', 'Burn_C','Burn_C','Burn_C','Burn_C', 'Control_D', 'Control_D','Control_D','Control_D','Control_D', 'Control_A', 'Control_A','Control_A','Control_A','Control_A', 'Burn_B', 'Burn_B','Burn_B','Burn_B','Burn_B', 'Burn_C', 'Burn_C','Burn_C','Burn_C','Burn_C', 'Control_D', 'Control_D','Control_D','Control_D','Control_D', 'Control_A', 'Control_A','Control_A','Control_A','Control_A')
roots_pcoa_vectors_metadata <- data.frame(Treatment_color = roots_treatment_pcoa_color, roots_pcoa_vectors_metadata)

tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/roots_16s/Results/16s_roots_PCOA.tiff", units = 'in', width = 10, height = 7, res = 200)
roots_pcoa_16s <- ggplot(roots_pcoa_scores, aes(PC1, PC2, color = roots_pcoa_vectors_metadata$Treatment_color))+
  geom_point(size = 2, aes(shape = year_permanova_roots))+
  labs(shape = 'Year')+
  ggtitle('Roots 16S')+
  xlab('Principal Axis 1 (16.36%)')+
  xlim(c(-0.6, 0.6))+
  ylab('Principal Axis 2 (9.68%)')+
  theme_linedraw()+
  #scale_color_discrete(name = 'Niches')+
  theme(plot.title = element_text(family = 'Arial', size=14, face="bold"))+
  theme(axis.title.x=element_text(family = 'Arial', size=10))+
  theme(axis.title.y=element_text(family = 'Arial', size=10))+
  theme(axis.text.x=element_text(family = 'Arial', size=10))+
  theme(axis.text.y=element_text(family = 'Arial', size=10))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 14))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))+
  scale_color_brewer(palette = 'Set1', name = 'Treatment')
dev.off()
```

# The purpose of this code is to do PERMANOVA on the WHOLE data

```{r}
# PERMANOVA without plot information
treatment_roots_whole <- c("Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control")

year_permanova_roots <- c("2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019")

roots_permanova_whole <- adonis(rare_roots_rel_abund_for_pcoa~treatment_roots_whole + year_permanova_roots + treatment_roots_whole * year_permanova_roots, method = 'bray', permutations = 10000)
roots_permanova_whole

roots_plot <- c("B","B","B","B","B","C","C","C","C","C","D","D","D","D","D","A","A","A","A","A","B","B","B","B","B","C","C","C","C","C","D","D","D","D","D","A","A","A","A","A")

#PERMANOVA with plot information
roots_permanova_whole_plot <- adonis(rare_roots_rel_abund_for_pcoa ~ treatment_roots_whole +  year_permanova_roots + roots_plot + treatment_roots_whole * year_permanova_roots * roots_plot, method = 'bray', permutations = 10000)
roots_permanova_whole_plot
```

#The purpose of this code chunk is to find differences in shannon alpha diversity across treatments.

```{r Differences in shannon index}
#We have to generate a new object for plot information
plot_tree_roots <- c("B_1", "B_2","B_3","B_4","B_5", "C_1", "C_2","C_3","C_4","C_5", "D_1", "D_2", "D_3", "D_4", "D_5", "A_1", "A_2","A_3","A_4","A_5","B_1", "B_2","B_3","B_4","B_5", "C_1", "C_2","C_3","C_4","C_5", "D_1", "D_2", "D_3", "D_4", "D_5", "A_1", "A_2","A_3","A_4","A_5")

#We will start with object 'roots_alpha_diversity' and add some metadata to it
roots_plot_shannon <- data.frame(Plot = plot_tree_roots, Treatment = treatment_roots_whole, Year = year_permanova_roots, roots_alpha_diversity)
colnames(roots_plot_shannon)[4] <- "Shannon_index"

#Install required packages if they are not already installed
if(!require(psych)){install.packages("psych")}
if(!require(nlme)){install.packages("nlme")}
if(!require(car)){install.packages("car")}
if(!require(multcompView)){install.packages("multcompView")}
if(!require(lsmeans)){install.packages("lsmeans")}
if(!require(ggplot2)){install.packages("ggplot2")}
if(!require(rcompanion)){install.packages("rcompanion")}

#Order factors by order in dataframe, otherwise, R will alphabetize them
roots_plot_shannon$Treatment <- factor(roots_plot_shannon$Treatment, levels = unique(roots_plot_shannon$Treatment))

#Convert the Year column to numeric
roots_plot_shannon$Year <- as.numeric(roots_plot_shannon$Year)

#Code for the analysis
library(nlme)
roots_model_alpha_diversity <- lme(Shannon_index ~ Treatment + Year + Treatment*Year, random = ~1|Plot, data = roots_plot_shannon, method = "REML")
library(car)
Anova(roots_model_alpha_diversity)

library(lsmeans)
roots_shannon_posthoc <- lsmeans(roots_model_alpha_diversity, ~Treatment)
roots_shannon_posthoc

library(multcomp)
cld(roots_shannon_posthoc, alpha = 0.05, Letters = letters, adjust = "tukey")

#Code for the post hoc analysis
roots_plot_shannon$treatment_year <- interaction(roots_plot_shannon$Treatment, roots_plot_shannon$Year)
roots_alpha_model <- lme(Shannon_index ~ treatment_year, random = ~1|Plot, correlation = corAR1(form = ~ Year | Plot, value = 0.4287), data = roots_plot_shannon, method = "REML")
Anova(roots_alpha_model)

#Let's perform the tukey test on roots alpha diversity
library(lsmeans)
roots_alpha_post_hoc <- lsmeans(roots_alpha_model, ~treatment_year, adjust = "tukey")
cld(roots_alpha_post_hoc, alpha = 0.05, Letters = letters, adjust = "tukey")

treatment_new_roots <- c("Unburned", "Unburned","Unburned", "Unburned","Unburned", "Unburned","Unburned", "Unburned","Unburned", "Unburned","Unburned", "Unburned","Unburned", "Unburned","Unburned", "Unburned","Unburned", "Unburned","Unburned", "Unburned", "Burned", "Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned", "Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned")
roots_plot_shannon_new <- data.frame(roots_plot_shannon, Treatment_new = treatment_new_roots)

#Order factors by order in dataframe, otherwise, R will alphabetize them
roots_plot_shannon_new$Treatment_new <- factor(roots_plot_shannon_new$Treatment_new, levels = unique(roots_plot_shannon_new$Treatment_new))

#Convert the Year column to numeric
roots_plot_shannon_new$Year <- as.numeric(roots_plot_shannon_new$Year)

library(lme4)
roots_plot_shannon_new_model <- lme4::lmer(Shannon_index ~ Treatment_new + Year + Treatment_new*Year, random = ~1|Plot, correlation = corAR1(form = ~ Year | Plot, value = 0.4287), data = roots_plot_shannon_new, method = "REML")
```

#The purpose of this code is find the differences in observed richness across treatments.

```{r Differences in Richness}
# RICHNESS
library(vegan)
roots_richness <- specnumber(rare_otu_roots)
roots_richness_metadata <- data.frame(Plot = plot_tree_roots, Year = year_permanova_roots, Treatment = treatment_roots_whole, Richness = roots_richness)

#Order factors by order in dataframe, otherwise, R will alphabetize them
roots_richness_metadata$Treatment <- factor(roots_richness_metadata$Treatment, levels = unique(roots_richness_metadata$Treatment))

#Make the Year column numeric
roots_richness_metadata$Year <- as.numeric(roots_richness_metadata$Year)

#Code for the analysis
library(nlme)
roots_model_richness <- lme(Richness ~ Treatment + Year + Treatment*Year, random = ~1|Plot, correlation = corAR1(form = ~ Year | Plot, value = 0.4287), data = roots_richness_metadata, method = "REML")
library(car)
Anova(roots_model_richness)

#Code for the post hoc analysis
roots_richness_metadata$treatment_year <- interaction(roots_richness_metadata$Treatment, roots_richness_metadata$Year)
roots_richness_model_tukey <- lme(Richness ~ treatment_year, random = ~1|Plot, correlation = corAR1(form = ~ Year | Plot, value = 0.4287), data = roots_richness_metadata, method = "REML")

#Let's perform the tukey test on roots alpha diversity
library(lsmeans)
roots_richness_post_hoc <- lsmeans(roots_richness_model_tukey, ~treatment_year, adjust = "tukey")
cld(roots_richness_post_hoc, alpha = 0.05, Letters = letters, adjust = "tukey")

#One thing that we forgot is to make charts for the richness. We will do that now. Let's create the data frame required.
roots_richness_df <- data.frame(roots_alpha_metadata, Richness = roots_richness)
roots_richness_df <- roots_richness_df[, -2]
roots_richness_df$Treatment <- as.character(roots_richness_df$Treatment)
roots_richness_df$Treatment <- factor(roots_richness_df$Treatment, levels = c('Pre_control_A', 'Post_control_A', 'Pre_control_D', 'Post_control_D', 'Pre_burn_B', 'Post_burn_B', 'Pre_burn_C', 'Post_burn_C'))

# This is the graph with legend
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/roots_16s/Results/16s_roots_richness.tiff", units = 'in', width = 10, height = 7, res = 200)
ggplot(roots_richness_df, aes(x = Treatment, y = Richness, fill = Treatment))+
  geom_boxplot(outlier.shape = NA)+
  ggtitle('Roots 16S Richness')+
  ylab('OTU Richness')+
  xlab('Treatment')+
  theme_linedraw()+
  theme(axis.title.x = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(plot.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.x = element_text(family = 'Times New Roman', size = 14, colour = 'black', face = 'bold', angle = 90))+
  theme(axis.title.y = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.y = element_text(family = 'Times New Roman', size = 14, face = 'bold'))+
  theme(legend.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(legend.text = element_text(size = 18, family = 'Times New Roman'))+
  scale_fill_brewer(palette = 'Set1')+
  guides(fill=guide_legend(title = 'Treatment'))
dev.off()

#Let's make the graph without legend
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/roots_16s/Results/16s_roots_richness_without_legend.tiff", units = 'in', width = 8, height = 7, res = 200)
roots_richness_chart_without_legend <- ggplot(roots_richness_df, aes(x = Treatment, y = Richness, fill = Treatment))+
  geom_boxplot(outlier.shape = NA)+
  ggtitle('Roots 16S Richness')+
  ylab('OTU Richness')+
  xlab('Treatment')+
  theme_linedraw()+
  theme(axis.title.x = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(plot.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.x = element_text(family = 'Times New Roman', size = 14, colour = 'black', face = 'bold', angle = 90))+
  theme(axis.title.y = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.y = element_text(family = 'Times New Roman', size = 14, face = 'bold'))+
  theme(legend.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(legend.text = element_text(size = 18, family = 'Times New Roman'))+
  theme(legend.position = 'none')+
  scale_fill_brewer(palette = 'Set1')+
  guides(fill=guide_legend(title = 'Treatment'))
dev.off()

#Now let's combine the alpha diversity and richness chart using ggarrange
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/roots_16s/Results/16s_roots_richness_and_shannon.tiff", units = 'in', height = 10, width = 8, res = 300)
ggarrange(roots_richness_chart_without_legend + theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()), roots_alpha_without_legend,  ncol = 1, labels = c("I", "II"), align = 'v', hjust = -3)
dev.off()
```

```{r Concise box plots}
#The purpose of this code chunk is to make the shannon diversity box plots in the concise form. We will start working with dataframe roots_alpha_diversity
roots_16s_alpha_diversity <- roots_alpha_diversity
colnames(roots_16s_alpha_diversity)[1] <- 'values'

#Note that the metadata that we wanna use now has already been created for pcoa plot so will use that metadata.
roots_treatment_pcoa_color
year_permanova_roots
roots_16s_alpha_concise <- data.frame(roots_16s_alpha_diversity, Treatment = roots_treatment_pcoa_color, Year = year_permanova_roots)
roots_16s_alpha_concise$Treatment <- as.character(roots_16s_alpha_concise$Treatment)
roots_16s_alpha_concise$Treatment <- factor(roots_16s_alpha_concise$Treatment, levels = c('Control_A', 'Control_D', 'Burn_B', 'Burn_C'))

roots_16s_alpha <- ggplot(data = roots_16s_alpha_concise, aes(x = Treatment, y = values)) + geom_boxplot(aes(fill = Year)) +
  ggtitle('Roots 16S')+
  xlab('Treatment')+
  ylab('Shannon Diversity Index')+
  theme_linedraw()+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  theme(axis.text.x=element_text(family = 'Arial', size=12))+
  theme(axis.text.y=element_text(family = 'Arial', size=12))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))+
  scale_color_brewer(palette = 'Set1', name = 'Treatment')

#We have to do the same for the richness graph. We will use roots_richness_df dataframe
roots_16s_richness <- roots_richness_df

#We wanna delete the existing Treatment column
roots_16s_richness <- subset(roots_16s_richness, select = -Treatment)

#Add the same metadata previously added
roots_16s_richness <- data.frame(roots_16s_richness, Treatment = roots_treatment_pcoa_color, Year = year_permanova_roots)
roots_16s_richness$Treatment <- as.character(roots_16s_richness$Treatment)
roots_16s_richness$Treatment <- factor(roots_16s_richness$Treatment, levels = c('Control_A', 'Control_D', 'Burn_B', 'Burn_C'))

roots_16s_richness_graph <- ggplot(data = roots_16s_richness, aes(x = Treatment, y = Richness)) + geom_boxplot(aes(fill = Year)) +
  #ggtitle('Roots 16S')+
  xlab('Treatment')+
  ylab('OTU Richness')+
  theme_linedraw()+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  scale_y_continuous(position = "right")+
  theme(axis.text.x=element_text(family = 'Arial', size=12))+
  theme(axis.text.y=element_text(family = 'Arial', size=12))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))+
  scale_color_brewer(palette = 'Set1', name = 'Treatment')
```

```{r Anova test on 2018 data}
# We will start with subsetting the roots_alpha_diversity dataframe
roots_alpha_2018 <- as.data.frame(roots_alpha_metadata[1:20, 2])
colnames(roots_alpha_2018)[1] <- 'alpha_diversity'

plot_roots_anova <- c("B", "B","B","B","B","C","C","C","C","C","D","D","D","D","D","A","A","A","A","A")
roots_alpha_2018 <- data.frame(Plot = plot_roots_anova, roots_alpha_2018)

roots_alpha_2018_anova <- aov(alpha_diversity ~ Plot, data = roots_alpha_2018)
summary(roots_alpha_2018_anova)
```

```{r Constrained Correspondence Analysis (CCA)}
roots_nutrients <- soils_nutrients
rare_otu_roots

#In the roots_nutrients dataframe, let's replace the underscores in rownames with nothing
rownames(roots_nutrients) <- gsub('_', '', rownames(roots_nutrients))

#We have to delete the last "1" in the rownames of roots_nutrients and replace it with "2" to match roots naming
rownames(roots_nutrients) <- gsub('1$', "2", rownames(roots_nutrients))

#let's merge these two dataframes using the rownames
roots_otu_nutrients <- merge(roots_nutrients, rare_otu_roots, by = 0)

#CCA and Model Selection. I then perform the CCA using cca function from vegan. Prior to performing the CCA, I remove the variables that are strongly positively correlated. I then perform the model selection using the ordistep function to determine which variables to retain in the model.

#Below I create a matrix that consists of only soil physicochemical properties
roots_physicochemical <- data.frame(roots_nutrients[, 3:15])

#I then calculate the correlations for each soil variable to evaluate multicollinearity within the data
roots_corrtest <- cor(roots_physicochemical, method = "pearson")

#I plot the correlations using the corrplot function
library(corrplot)
corrplot(roots_corrtest, method = "number")

#Looking at the correlation matrix we can keep pH, S, Zn, Mn and NH.N. All other variables are highly correlated with these major drivers of the bacterial community composition.
roots_important_physicochemical <- data.frame(roots_nutrients[ , c(2, 4, 11, 13, 15)])

#Let's do the correlation test on the above matrix too
roots_imp_corrtest <- cor(roots_important_physicochemical, method = "pearson")
corrplot(roots_imp_corrtest, method = "number") #Looks good.

#I then create a matrix that consists only of the OTU table (must be numeric only)
roots_prop_otu <- roots_otu_nutrients[, 17:9390]

library(vegan)
#Set seed to get consistent point placement in ordination space
set.seed(786)

#Perform CCA with cca function from vegan. The first CCA is the intercept only and the second is the full model. These are needed for the model selection in subsequent steps.
roots_cca_intercept <- cca(roots_prop_otu ~ 1, data = roots_important_physicochemical)
roots_ccaplot <- cca(roots_prop_otu ~ ., data = roots_important_physicochemical)

#I evaluate multicollinearity one more time using the vif.cca function
vif.cca(roots_ccaplot)

#I then perform forward model selection using the ordistep function. An alternative method is to both directions or use the ordiR2step function.
roots_cca_model <- ordistep(roots_cca_intercept, scope = roots_ccaplot, permutations = how(nperm = 999), trace = TRUE, direction = "forward") 

#To determine if soil physicochemical properties significantly explains the bacterial community perform permutation test with anova function. By including the 'by' argument we can identify significance of axes and model terms.
anova(roots_cca_model, steps = 10000) #No constrained component

#By axis
anova(roots_cca_model, steps = 10000, by = "axis")

#By terms
anova(roots_cca_model, steps = 10000, by = "terms")

# To obtain the information on axes, vectors etc. obtain CCA summary
roots_cca_model_summary <- summary(roots_cca_model)

#CCA Plot. Then to plot the CCA, I use the ggplot function. To use ggplot I need to first pull out the site scores for each sample, arrow coordinates, and generate ellipses.

#Since vegna does not feed directly into ggplot you will need to extract scores for sites and vectors to make CCAs in gggplot. Begin by extracting site scores (these will be the x and y coordinates for each sample)
roots_cca_site_scores <- as.data.frame(scores(roots_cca_model_summary, display = 'sites'))

#Generate some metadata to add to the dataframe
year_cca_roots <- c("2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019")

#Create a new dataframe that includes the site scores from above with metadata from your study
roots_cca_graph <- data.frame(roots_cca_site_scores, CCA1 = roots_cca_site_scores$CCA1, CCA2 = roots_cca_site_scores$CCA2, Treatment = treatment_roots_nutrients, Year = year_cca_roots)

#To create a biplot you need to extract vector scores as below. To extract vector scores you need to use the object that contains the summary for CCA
roots_cca_arrows <- as.data.frame(roots_cca_model_summary$biplot)

#Then you need to create an object that contains the labels for the soil arrows and the coordinates for the soil arrows
roots_cca_labels <- labels(roots_cca_arrows)

#Here I am pulling out the arrow scores that are significant
roots_cca_arrows_ph1 <- roots_cca_arrows$CCA1[[c(1)]]
roots_cca_arrows_ph2 <- roots_cca_arrows$CCA2[[c(1)]]
roots_cca_arrows_ph <- data.frame(CCA1 = roots_cca_arrows_ph1, CCA2 = roots_cca_arrows_ph2)

roots_cca_arrows_mn1 <- roots_cca_arrows$CCA1[[c(2)]]
roots_cca_arrows_mn2 <- roots_cca_arrows$CCA2[[c(2)]]
roots_cca_arrows_mn <- data.frame(CCA1 = roots_cca_arrows_mn1, CCA2 = roots_cca_arrows_mn2)

#Here you determine the scaling factor for geom segment and geom point
roots_cca_arrowmult <- ordiArrowMul(roots_cca_model_summary, display = "sites")

#This is where you make confidence ellipses. I don't know what all the code means. Just know where to plug in my objects
roots_cca_ellipse <- ordiellipse(roots_cca_site_scores, roots_cca_graph$Treatment, display = "sites", kind = "sd", draw = "none")

df_ell_cca <- data.frame()
for(g in levels(roots_cca_graph$Treatment)){
  df_ell_cca <- rbind(df_ell_cca, cbind(as.data.frame(with(roots_cca_graph[roots_ccaplot$Treatment == g, ], vegan:::veganCovEllipse(roots_cca_ellipse[[g]]$cov, roots_cca_ellipse[[g]]$center, roots_cca_ellipse[[g]]$scale))), Treatment = g))
}

roots_cca_color <- length(unique(df_ell_cca$Treatment))
#Plot CCA in ggplot
library(ggplot2)

roots_cca_chart <- ggplot(roots_cca_graph, aes(CCA1, CCA2, colour = Treatment))+
  geom_point(size = 3)+
  geom_path(data = df_ell_cca, aes(x = CCA1, y = CCA2, colour = Treatment), size = 1, linetype = 2)+
  geom_text(data = roots_cca_arrows_ph, aes(x = 2*CCA1, y = 2*CCA2, label = 'pH', fontface = "bold"), inherit.aes = FALSE)+
  geom_segment(data = roots_cca_arrows_ph, aes(x = 0, xend = 1.742239*CCA1, y = 0, yend = 1.742239 * CCA2), arrow = arrow(length = unit(0.25, "cm")), size = 0.5, linetype = 1, color = "black")+
  geom_text(data = roots_cca_arrows_mn, aes(x = 2*CCA1, y = 2*CCA2, label = 'Mn', fontface = "bold"), inherit.aes = FALSE)+
  geom_segment(data = roots_cca_arrows_mn, aes(x = 0, xend = 1.742239*CCA1, y = 0, yend = 1.742239 * CCA2), arrow = arrow(length = unit(0.25, "cm")), size = 0.5, linetype = 1, color = "black")+
  scale_fill_manual(values = colorRampPalette(brewer.pal(7, 'Accent'))(roots_cca_color))+
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(axis.title.x=element_text(size=14, face="bold"))+
  scale_x_continuous(breaks=c(-3,-1.5,0,1.5,3),limits=c(-3,3))+
  scale_y_continuous(breaks=c(-3,-1.5,0,1.5,3),limits=c(-3,3))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))
  
roots_cca_chart
```

#The purpose of this code chunk is to prepare alpha diversity and richness boxplots without plot information

```{r Box plots without plot}
#The purpose of this code chunk is to make the shannon diversity box plots in the concise form. We will start working with dataframe roots_alpha_diversity
roots_16s_alpha_diversity <- roots_alpha_diversity
colnames(roots_16s_alpha_diversity)[1] <- 'values'

#Create metadata
roots_treatment <- c("Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Control", "Control","Control","Control","Control","Control","Control","Control","Control","Control","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Control", "Control","Control","Control","Control","Control","Control","Control","Control","Control")
year_permanova_roots
roots_16s_alpha_without_plot <- data.frame(roots_16s_alpha_diversity, Treatment = roots_treatment, Year = year_permanova_roots)
roots_16s_alpha_without_plot$Treatment <- as.character(roots_16s_alpha_without_plot$Treatment)
roots_16s_alpha_without_plot$Treatment <- factor(roots_16s_alpha_without_plot$Treatment, levels = c("Control", "Burned"))

roots_16s_alpha_graph_without_plot <- ggplot(data = roots_16s_alpha_without_plot, aes(x = Treatment, y = values)) + geom_boxplot(aes(fill = Year), outlier.shape = NA) +
  ggtitle('Roots 16S')+
  xlab('Treatment')+
  ylab('Shannon Diversity Index')+
  theme_linedraw()+
  geom_text(data=NULL,aes(x=0.81, y=5.05, label="A"))+
  geom_text(data=NULL,aes(x=1.19, y=5.1, label="A"))+ 
  geom_text(data=NULL,aes(x=1.81, y=4.8, label="A"))+ 
  geom_text(data=NULL,aes(x=2.19, y=5.25, label="A"))+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  theme(axis.text.x=element_text(family = 'Arial', size=12))+
  theme(axis.text.y=element_text(family = 'Arial', size=12))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))+
  scale_color_brewer(palette = 'Set1', name = 'Treatment')

roots_16s_alpha_graph_without_plot

#We have to do the same for the richness graph. We will use roots_richness_df dataframe
roots_16s_richness <- roots_richness_df

#We wanna delete the existing Treatment column
roots_16s_richness <- subset(roots_16s_richness, select = -Treatment)

#Add the same metadata previously added
roots_16s_richness_without_plot <- data.frame(roots_16s_richness, Treatment = roots_treatment, Year = year_permanova_roots)
roots_16s_richness_without_plot$Treatment <- as.character(roots_16s_richness_without_plot$Treatment)
roots_16s_richness_without_plot$Treatment <- factor(roots_16s_richness_without_plot$Treatment, levels = c("Control", "Burned"))

roots_16s_richness_graph_without_plot <- ggplot(data = roots_16s_richness_without_plot, aes(x = Treatment, y = Richness)) + geom_boxplot(aes(fill = Year), outlier.shape = NA) +
  #ggtitle('roots 16S')+
  xlab('Treatment')+
  ylab('OTU Richness')+
  geom_text(data=NULL,aes(x=0.82, y=675, label="AB"))+
  geom_text(data=NULL,aes(x=1.18, y=770, label="AB"))+ 
  geom_text(data=NULL,aes(x=1.82, y=475, label="A"))+ 
  geom_text(data=NULL,aes(x=2.18, y=1075, label="B"))+
  theme_linedraw()+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  scale_y_continuous(position = "right")+
  theme(axis.text.x=element_text(family = 'Arial', size=12))+
  theme(axis.text.y=element_text(family = 'Arial', size=12))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))+
  scale_color_brewer(palette = 'Set1', name = 'Treatment')

roots_16s_richness_graph_without_plot
```

#The purpose of this chunk is to perform the indicator species analysis on roots data

```{r Indicator Species Analysis}
#We will use data frame rare_otu_roots_no_singletons and make a new vector to use as site classification
treatment_roots_indicator <- c("Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019")

#We are pretty much ready to run the analysis on roots data
library(indicspecies)
set.seed(786)
roots_indval <- multipatt(rare_otu_roots_no_singletons, treatment_roots_indicator, func = "IndVal.g", duleg = TRUE, control = how(nperm = 10000))

sink(file = "/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Roots_16s/Tables_from_R/roots_16s_indicator_species.txt")
summary(roots_indval, indvalcomp = TRUE)
sink()

library(readxl)
roots_indicators <- read_excel("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Roots_16s/Tables_from_R/16s_roots_indicator.xlsx", col_names = TRUE)
roots_indicators <- as.data.frame(roots_indicators)
rownames(roots_indicators) <- roots_indicators[, 2]
roots_indicators_test <- roots_indicators[, -2]

#Subset the dataframes based on treatment
roots_indic_control_2018 <- subset(roots_indicators_test, roots_indicators_test$Treatment == 'Control 2018')
roots_indic_control_2019 <- subset(roots_indicators_test, roots_indicators_test$Treatment == 'Control 2019')
roots_indic_burned_2018 <- subset(roots_indicators_test, roots_indicators_test$Treatment == 'Burned 2018')
roots_indic_burned_2019 <- subset(roots_indicators_test, roots_indicators_test$Treatment == 'Burned 2019')

#Keep first ten rows
roots_indic_control_2018_imp <- roots_indic_control_2018[1:10, ]
roots_indic_control_2019_imp <- roots_indic_control_2019[1:10, ]
roots_indic_burned_2018_imp <- roots_indic_burned_2018[1:10, ]
roots_indic_burned_2019_imp <- roots_indic_burned_2019[1:10, ]

roots_indic_imp <- rbind(roots_indic_control_2018_imp, roots_indic_control_2019_imp, roots_indic_burned_2018_imp, roots_indic_burned_2019_imp)

#Create one more dataframe only with OTU and treatment info
roots_indic_treatment <- as.data.frame(roots_indic_imp[, 1])
colnames(roots_indic_treatment) <- "Treatment"
rownames(roots_indic_treatment) <- rownames(roots_indic_imp)

roots_indic_treatment$Treatment <- factor(roots_indic_treatment$Treatment, levels = c("Control 2018", "Control 2019", "Burned 2018", "Burned 2019"))

library(RColorBrewer)
roots_ind_heatmap <- pheatmap(roots_indic_imp[, 2:4], 
                              treeheight_row = 0, 
                              treeheight_col = 0,
                              annotation_row = roots_indic_treatment,
                              color = colorRampPalette(brewer.pal(8, "Blues"))(100),
                              cluster_rows = F,
                              gaps_row = c(10, 20, 30))
roots_ind_heatmap


#To find out the identity of the indicator species we have to merge indicator species dataframe to roots taxonomy dataframe
library(tidyr)
roots_indic_taxonomy <- merge(roots_indic_imp, roots_otu_taxonomy, by = 0)

#Subset by treatment so that we get OTUs which belong to "Burned 2019"
roots_indic_tax_burned_2019 <- roots_indic_taxonomy[roots_indic_taxonomy$Treatment == "Burned 2019", ]
roots_indic_tax_burned_2019 <- roots_indic_tax_burned_2019[order( - roots_indic_tax_burned_2019$Indicator_value), ]
```

# This is hopefully the last way I am making my boxplots from richness and alpha diversity. Let's start from a fresh dataset

```{r}
# Use roots_16s_alpha_without_plot and make a new dataframe out of it
alpha_roots <- roots_16s_alpha_without_plot

# Prepare a new vector
alpha_roots_treatment <- c("Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019", "Control 2019", "Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019")

#Add the above vector to alpha_roots
alpha_roots <- data.frame(alpha_treatment = alpha_roots_treatment, alpha_roots)

#Delete Treatment and year column
alpha_roots_clean <- alpha_roots[, -c(3, 4)]

#Change the column name alpha_treatment to simply Treatment
colnames(alpha_roots_clean)[1] <- "Treatment"

#Convert the variable Treatment to factor variable
alpha_roots_clean$Treatment <- factor(alpha_roots_clean$Treatment, levels = c("Control 2018", "Control 2019", "Pre-burned 2018", "Post-burned 2019"))

#Let's make the boxplot
library(ggplot2)
p_roots <- ggplot(alpha_roots_clean, aes(x = Treatment, y = values, fill = Treatment)) +
  geom_boxplot(outlier.shape = NA)+
  theme_linedraw()+
  scale_fill_manual(values = alpha(c("lightgreen","springgreen4","bisque1","firebrick3"), 0.9))+
  ggtitle('Roots 16S')+
  #xlab('Treatment')+
  #ylab('Shannon Diversity Index')+
  theme(legend.title = element_text(face = "bold", size = 14),
        legend.text = element_text(size = 14))+
  theme(plot.title = element_text(face = "bold", size = 14))+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  theme(axis.text.x = element_blank())
  #theme(axis.text.x=element_text(family = 'Arial', size=12, angle = 30, vjust = 0.6))
  #theme(axis.text.y=element_text(family = 'Arial', size=12))
  #theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  #theme(legend.text = element_text(size = 14, family = 'Arial'))
  #theme(plot.margin = unit(c(0,0,0,0), "lines"))

p_roots

#Have to do the same for richness. Let's start with the dataframe roots_16s_richness and add Treatment column
roots_rich <- data.frame(roots_16s_richness, Treatment = alpha_roots_treatment)
roots_rich$Treatment <- factor(roots_rich$Treatment, levels = c("Control 2018", "Control 2019", "Pre-burned 2018", "Post-burned 2019"))

#Let's make the boxplot
library(ggplot2)
r_roots <- ggplot(roots_rich, aes(x = Treatment, y = Richness, fill = Treatment)) +
  geom_boxplot(outlier.shape = NA)+
  theme_linedraw()+
  scale_fill_manual(values = alpha(c("lightgreen","springgreen4","bisque1","firebrick3"), 0.9))+
  #ggtitle('Roots 16S')+
  #xlab('Treatment')+
  #ylab('Shannon Diversity Index')+
  theme(legend.title = element_text(face = "bold", size = 14),
        legend.text = element_text(size = 14))+
  theme(plot.title = element_text(face = "bold", size = 14))+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  theme(axis.text.x = element_blank())+
  #theme(axis.text.y=element_text(family = 'Arial', size=12))+
  scale_y_continuous(position = "right")
  #theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  #theme(legend.text = element_text(size = 14, family = 'Arial'))
  #theme(plot.margin = unit(c(0,0,0,0), "lines"))

r_roots
```

#T.test by treatment

```{r}
#Let's use the roots_plot_shannon
roots_alpha_diversity_treatment <- roots_plot_shannon[, c(2:4)]
roots_alpha_diversity_treatment$X <- rownames(roots_alpha_diversity_treatment)

roots_alpha_diversity_treatment$X<-sub("18","",roots_alpha_diversity_treatment$X)
roots_alpha_diversity_treatment$X<-sub("19","",roots_alpha_diversity_treatment$X)
 
library(tidyr)
roots_alpha_diversity_wide<-pivot_wider(roots_alpha_diversity_treatment,names_from=Year,values_from=Shannon_index)
 
roots_alpha_diversity_wide$yeardiffs<-roots_alpha_diversity_wide$`2`- roots_alpha_diversity_wide$`1`
 
t.test.year.diffs_roots<-t.test(yeardiffs~Treatment,roots_alpha_diversity_wide)
t.test.year.diffs_roots
summary(t.test.year.diffs)
plot(t.test.year.diffs)
hist(roots_alpha_diversity_wide$yeardiffs)

#Richness
roots_richness_treatment <- data.frame(roots_rich, Treatment = roots_alpha_diversity_treatment$Treatment, Year = year_permanova_roots)

#Delete the treatment column
roots_richness_treatment <- roots_richness_treatment[, -2]
colnames(roots_richness_treatment)[2] <- "Treatment"

roots_richness_treatment$Sample <- rownames(roots_richness_treatment)
roots_richness_treatment$Sample <- sub("18", "", roots_richness_treatment$Sample)
roots_richness_treatment$Sample <- sub("19", "", roots_richness_treatment$Sample)

library(tidyr)
roots_rich_wide <- pivot_wider(roots_richness_treatment, names_from = Year, values_from = Richness)
roots_rich_wide$yeardiffs <- roots_rich_wide$`2019` - roots_rich_wide$`2018`
hist(roots_rich_wide$yeardiffs)

t_test_roots_rich <- t.test(yeardiffs ~ Treatment, roots_rich_wide)
t_test_roots_rich
```

#T-test by year

```{r}
library(tidyr)
roots_alpha_t_year <- roots_alpha_diversity_treatment[, -1]
roots_alpha_t_wide <- pivot_wider(roots_alpha_t_year, names_from = Year, values_from = Shannon_index)

t_roots_alpha_year <- t.test(roots_alpha_t_wide$`2`, roots_alpha_t_wide$`1`, paired = TRUE, alternative = "two.sided")
t_roots_alpha_year

#Richness
roots_richness_t <- roots_richness_treatment
roots_richness_t <- roots_richness_t[, -2]
roots_richness_wide <- pivot_wider(roots_richness_t, names_from = Year, values_from = Richness)

roots_richness_t_year <- t.test(roots_richness_wide$`2019`, roots_richness_wide$`2018`, paired = TRUE, alternative = "two.sided")
roots_richness_t_year
```
