---
title: "16s_leaves"
author: "Beant Kapoor"
date: "3/9/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(echo=FALSE, warning=FALSE, message=FALSE)
```

### General Comment - Objects in this file contain the name 'otu' however they represent ASV in this case.

```{r Import Data and Subset}
#We already have imported the otu file and taxonomy file from DADA2 to this R project/environment. We just need to import the metadata for leaves niche and subset the data accordingly.
leaves_metadata <- read.table("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/metadata/16s_leaves.txt", header = TRUE, fill = TRUE, strip.white = TRUE)
leaves_sample_names <- leaves_metadata$Sample

#Here I use the subset function to pull out just the samples belonging to leaves
leaves_otu <- subset(whole_otu, rownames(whole_otu) %in% leaves_sample_names)
```

```{r Data Clean-up}
#Let's remove the columns which are not needed
leaves_otu_only <- leaves_otu[, 3:170189]

#Here I am removing any OTU that is not present in any samples. What the below function is doing is summing the columns (OTUs) and any OTU that equals 0 is removed from the dataset. This will hopefully speed up downstream analyses.
leaves_pure <- leaves_otu_only[, colSums(leaves_otu_only) > 0]
```

```{r Rarefaction}
#Below I create the rarefaction curves for each sample using the rarecurve function from vegan. 
library(vegan)
rare_curve_leaves <- rarecurve(leaves_pure)

#I then create an object that spans the range of the OTU richness. This will be used to plot the vertical line in my ggplot rarefaction curve. 
f_leaves <- (0:2000)

#I then create objects containing the x and y coordinates values for each stratum.
sequences_leaves40<-attr(rare_curve_leaves[[40]],which="Subsample")
sequences_leaves39<-attr(rare_curve_leaves[[39]],which="Subsample")
sequences_leaves38<-attr(rare_curve_leaves[[38]],which="Subsample")
sequences_leaves37<-attr(rare_curve_leaves[[37]],which="Subsample")
sequences_leaves36<-attr(rare_curve_leaves[[36]],which="Subsample")
sequences_leaves35<-attr(rare_curve_leaves[[35]],which="Subsample")
sequences_leaves34<-attr(rare_curve_leaves[[34]],which="Subsample")
sequences_leaves33<-attr(rare_curve_leaves[[33]],which="Subsample")
sequences_leaves32<-attr(rare_curve_leaves[[32]],which="Subsample")
sequences_leaves31<-attr(rare_curve_leaves[[31]],which="Subsample")
sequences_leaves30<-attr(rare_curve_leaves[[30]],which="Subsample")
sequences_leaves29<-attr(rare_curve_leaves[[29]],which="Subsample")
sequences_leaves28<-attr(rare_curve_leaves[[28]],which="Subsample")
sequences_leaves27<-attr(rare_curve_leaves[[27]],which="Subsample")
sequences_leaves26<-attr(rare_curve_leaves[[26]],which="Subsample")
sequences_leaves25<-attr(rare_curve_leaves[[25]],which="Subsample")
sequences_leaves24<-attr(rare_curve_leaves[[24]],which="Subsample")
sequences_leaves23<-attr(rare_curve_leaves[[23]],which="Subsample")
sequences_leaves22<-attr(rare_curve_leaves[[22]],which="Subsample")
sequences_leaves21<-attr(rare_curve_leaves[[21]],which="Subsample")
sequences_leaves20<-attr(rare_curve_leaves[[20]],which="Subsample")
sequences_leaves19<-attr(rare_curve_leaves[[19]],which="Subsample")
sequences_leaves18<-attr(rare_curve_leaves[[18]],which="Subsample")
sequences_leaves17<-attr(rare_curve_leaves[[17]],which="Subsample")
sequences_leaves16<-attr(rare_curve_leaves[[16]],which="Subsample")
sequences_leaves15<-attr(rare_curve_leaves[[15]],which="Subsample")
sequences_leaves14<-attr(rare_curve_leaves[[14]],which="Subsample")
sequences_leaves13<-attr(rare_curve_leaves[[13]],which="Subsample")
sequences_leaves12<-attr(rare_curve_leaves[[12]],which="Subsample")
sequences_leaves11<-attr(rare_curve_leaves[[11]],which="Subsample")
sequences_leaves10<-attr(rare_curve_leaves[[10]],which="Subsample")
sequences_leaves9<-attr(rare_curve_leaves[[9]],which="Subsample")
sequences_leaves8<-attr(rare_curve_leaves[[8]],which="Subsample")
sequences_leaves7<-attr(rare_curve_leaves[[7]],which="Subsample")
sequences_leaves6<-attr(rare_curve_leaves[[6]],which="Subsample")
sequences_leaves5<-attr(rare_curve_leaves[[5]],which="Subsample")
sequences_leaves4<-attr(rare_curve_leaves[[4]],which="Subsample")
sequences_leaves3<-attr(rare_curve_leaves[[3]],which="Subsample")
sequences_leaves2<-attr(rare_curve_leaves[[2]],which="Subsample")
sequences_leaves1<-attr(rare_curve_leaves[[1]],which="Subsample")

#Here I plot the rarefaction curves. 

library(ggplot2)
library(ggpubr)

rarefaction_chart_leaves <- ggplot() +
  geom_line(aes(x=sequences_leaves40,y=rare_curve_leaves[[40]]), size=1)+
  geom_line(aes(x=sequences_leaves39,y=rare_curve_leaves[[39]]), size=1)+
  geom_line(aes(x=sequences_leaves38,y=rare_curve_leaves[[38]]),size=1)+
  geom_line(aes(x=sequences_leaves37,y=rare_curve_leaves[[37]]),size=1)+
  geom_line(aes(x=sequences_leaves36,y=rare_curve_leaves[[36]]),size=1)+
  geom_line(aes(x=sequences_leaves35,y=rare_curve_leaves[[35]]),size=1)+
  geom_line(aes(x=sequences_leaves34,y=rare_curve_leaves[[34]]), size=1)+
  geom_line(aes(x=sequences_leaves33,y=rare_curve_leaves[[33]]), size=1)+
  geom_line(aes(x=sequences_leaves32,y=rare_curve_leaves[[32]]),size=1)+
  geom_line(aes(x=sequences_leaves31,y=rare_curve_leaves[[31]]),size=1 )+
  geom_line(aes(x=sequences_leaves30,y=rare_curve_leaves[[30]]),size=1)+
  geom_line(aes(x=sequences_leaves29,y=rare_curve_leaves[[29]]),size=1)+
  geom_line(aes(x=sequences_leaves28,y=rare_curve_leaves[[28]]),size=1)+
  geom_line(aes(x=sequences_leaves27,y=rare_curve_leaves[[27]]),size=1)+
  geom_line(aes(x=sequences_leaves26,y=rare_curve_leaves[[26]]),size=1)+
  geom_line(aes(x=sequences_leaves25,y=rare_curve_leaves[[25]]), size=1)+
  geom_line(aes(x=sequences_leaves24,y=rare_curve_leaves[[24]]),size=1)+
  geom_line(aes(x=sequences_leaves23,y=rare_curve_leaves[[23]]),size=1)+
  geom_line(aes(x=sequences_leaves22,y=rare_curve_leaves[[22]]),size=1)+
  geom_line(aes(x=sequences_leaves21,y=rare_curve_leaves[[21]]),size=1)+
  geom_line(aes(x=sequences_leaves20,y=rare_curve_leaves[[20]]),size=1)+
  geom_line(aes(x=sequences_leaves19,y=rare_curve_leaves[[19]]), size=1)+
  geom_line(aes(x=sequences_leaves18,y=rare_curve_leaves[[18]]),size=1)+
  geom_line(aes(x=sequences_leaves17,y=rare_curve_leaves[[17]]),size=1)+
  geom_line(aes(x=sequences_leaves16,y=rare_curve_leaves[[16]]),size=1)+
  geom_line(aes(x=sequences_leaves15,y=rare_curve_leaves[[15]]),size=1)+
  geom_line(aes(x=sequences_leaves14,y=rare_curve_leaves[[14]]), size=1)+
  geom_line(aes(x=sequences_leaves13,y=rare_curve_leaves[[13]]), size=1)+
  geom_line(aes(x=sequences_leaves12,y=rare_curve_leaves[[12]]),size=1)+
  geom_line(aes(x=sequences_leaves11,y=rare_curve_leaves[[11]]),size=1 )+
  geom_line(aes(x=sequences_leaves10,y=rare_curve_leaves[[10]]),size=1)+
  geom_line(aes(x=sequences_leaves9,y=rare_curve_leaves[[9]]),size=1)+
  geom_line(aes(x=sequences_leaves8,y=rare_curve_leaves[[8]]),size=1)+
  geom_line(aes(x=sequences_leaves7,y=rare_curve_leaves[[7]]),size=1)+
  geom_line(aes(x=sequences_leaves6,y=rare_curve_leaves[[6]]),size=1)+
  geom_line(aes(x=sequences_leaves5,y=rare_curve_leaves[[5]]), size=1)+
  geom_line(aes(x=sequences_leaves4,y=rare_curve_leaves[[4]]),size=1)+
  geom_line(aes(x=sequences_leaves3,y=rare_curve_leaves[[3]]),size=1)+
  geom_line(aes(x=sequences_leaves2,y=rare_curve_leaves[[2]]),size=1)+
  geom_line(aes(x=sequences_leaves1,y=rare_curve_leaves[[1]]),size=1)+
  geom_vline(xintercept = 1139, linetype = "dashed")+
  ylim(0, 1500)+
  xlim(0, 28000)+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
       panel.border=element_rect(color="black", size=1, fill=NA))+
 theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+ 
labs(x=("Sequencing Depth"), y=expression(bold(paste("OTU Richness", sep=""))))+
  labs(title = 'Leaves 16S')
rarefaction_chart_leaves

ggsave(filename = 'leaves_16s_rarefaction_curve.png', path = "/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Results/Leaves_16s", dpi = 300, width = 5, height = 5, units = 'in')

#Here we are rarefying the data using the rrarefy function from vegan. I first use the sort function to determine the lowest number of sequences in a sample. 
sort(rowSums(leaves_pure)) #1139

#Then using the rrarefy function from vegan, I rarefy my samples. The rarefaction cutoff or "sample" value is determined by evaluating my rarefaction curves and choosing the number of sequences that minimizes sample loss but maximizes sampling depth.
rare_otu_leaves <- as.data.frame(rrarefy(leaves_pure, sample = 1139)) #In this case we are not losing any samples therefore we do not need to perform any subsetting.

#I am now removing OTUs present in the OTU table that were only present in samples removed following rarefaction. These OTUs will have column sums of 0.
rare_otu_leaves <- rare_otu_leaves[, colSums(rare_otu_leaves) > 0]

#The following line is for saving the rarefied OTU table. This file is later reloaded to be used in downstream analyses. This is done to maintain consistency of results because rarefying generates slightly different results each time.
write.table(rare_otu_leaves, "/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Leaves_16s/Tables_generated_from_R/16s_leaves_rarefied_otu.shared", sep = '\t', row.names = TRUE, col.names = TRUE)

#I am now removing singleton OTUs (OTUs with only 1 representative sequence following rarefaction). This is to limit the effects of rare species on our distance matrices used during ordination.  
rare_otu_leaves_no_singletons <- rare_otu_leaves[, colSums(rare_otu_leaves) > 1]

#The following line is for saving the rarefied OTU table with singletons removed. This file is later reloaded to be used in downstream analyses. This is done to maintain consistency of results because rarefying generates slightly different results each time. 
write.table(rare_otu_leaves_no_singletons, "/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Leaves_16s/Tables_generated_from_R/16s_leaves_rarefied_no_singletons.shared", sep = '\t', row.names = TRUE, col.names = TRUE)
```

```{r Relative abundance chart - Phylum}
#Now that samples have been rarefied I can perform alpha and beta diversity analyses on my data sets. To begin I first reformat my OTU table so that I can do some filtering and add taxonomy information.
rare_otu_leaves_no_singletons <- read.table("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Leaves_16s/Tables_from_R/16s_leaves_rarefied_no_singletons.shared", header = TRUE, sep = '\t')

rare_otu_leaves <- read.table("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Leaves_16s/Tables_from_R/16s_leaves_rarefied_otu.shared", header = TRUE, sep = "\t")

#I now transpose my OTU table so that I can begin to add in taxonomy information.
trans_otu_leaves <- as.data.frame(t(rare_otu_leaves_no_singletons))

#The below function is asking specifically for the OTU IDs in the leaves OTU table. These are the row names. I will use these to subset the taxonomy file.
leaves_otu_names <- rownames(trans_otu_leaves)
leaves_taxonomy <- subset(taxonomy, rownames(taxonomy) %in% leaves_otu_names)

#Now I am subsetting the new taxonomy file to include only the taxonomy information stored in column 2. 
leaves_taxonomy_only <- leaves_taxonomy[, 2]

#Below I am using the separate function from tidyr to split the taxonomy strings into columns by semi-colon so that I can rename the OTUs to give them more meaning for downstream analyses.
library(tidyr)

#The SILVA database only provides classifications up to genus level, new columns are created only up to Genus level.
taxonomy_labels<-c("kingdom","phylum","class","order","family","genus")

#Below I create a new data frame with a column containing taxonomy information.
leaves_otu_taxonomy <- data.frame(taxonomy = leaves_taxonomy_only, trans_otu_leaves)

#I then separate the taxonomy strings into new columns labeled with the taxonomy labels saved in the taxonomy_labels object. Strings are being separated based on the presence of semi-colons.
leaves_otu_taxonomy <- separate(leaves_otu_taxonomy, col = taxonomy, into = taxonomy_labels, sep = ';')

#Below I create a data frame that consists of phylum assignment and the rarefied OTU table.
leaves_otu_phylum <- data.frame(phylum = leaves_otu_taxonomy$phylum, trans_otu_leaves)

#I then remove the assignment confidence values contained in parentheses found in each taxonomic string using the sub function.
leaves_otu_phylum$phylum <- sub("\\(.*)","",x = leaves_otu_phylum$phylum)

#The following code is for the generation of phylum relative abundance stacked bar charts. I first sum each OTU by the phylum it belongs to. 
library(plyr)
leaves_summed_otu_phylum <- as.data.frame(ddply(leaves_otu_phylum, .(phylum), colwise(sum)))
rownames(leaves_summed_otu_phylum) <- make.names(leaves_summed_otu_phylum$phylum)
leaves_summed_otu_phylum <- leaves_summed_otu_phylum[, 2:41]

#I then transpose the data frame so that sample names are now row names and column names are OTU names.
trans_leaves_otu_summed_phylum <- as.data.frame(t(leaves_summed_otu_phylum))
rownames(trans_leaves_otu_summed_phylum) <- sub('X', '', x = rownames(trans_leaves_otu_summed_phylum))

#Read in a new metadata file for leaves to make relative abundance charts
leaves_metadata_for_rel_abund <- read_excel("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/metadata/16s_metadata_without_plot.xlsx", sheet = 5)
trans_leaves_phylum_metadata <- merge(leaves_metadata_for_rel_abund, trans_leaves_otu_summed_phylum, by.x = 1, by.y = 0)

#Then we can delete the column with the sample names as it is not needed
trans_leaves_phylum_metadata <- trans_leaves_phylum_metadata[, 2:20]

#I then sum each OTU by Treatment column and subset to exclude sample names. This is because the following functions only work on matrices containing numeric data.
leaves_phylum_summed_treatment <- as.data.frame(ddply(trans_leaves_phylum_metadata, .(Treatment), colwise(sum)))
leaves_phylum <- leaves_phylum_summed_treatment[, 2:19]

#I am now creating an other category. Other includes those OTUs belonging to a phylum that comprises less than 1% of the total community. This new object is reffered to as phylum_others.
leaves_phylum_others <- leaves_phylum[, colSums(leaves_phylum)/sum(leaves_phylum) <= 0.01]
leaves_phylum_others_sum <- rowSums(leaves_phylum_others)

#I then create an object that contains all of the phyla that comprise more than 1% of the total community.
leaves_phylum_regular <- leaves_phylum[, colSums(leaves_phylum)/sum(leaves_phylum) > 0.01]

#I then create a dataframe containing the treatment information, the Other column and the remaining phyla
leaves_phylum_reg_others <- data.frame(Treatment = leaves_phylum_summed_treatment$Treatment, leaves_phylum_regular, Others = leaves_phylum_others_sum)

#These values are then converted to relative abundance using the decostand function from vegan. 
library(vegan)
leaves_phylum_rel_abund <- decostand(leaves_phylum_reg_others[, 2:9], method = 'total')
leaves_phylum_rel_abund <- data.frame(Treatment = leaves_phylum_reg_others$Treatment, leaves_phylum_rel_abund)

#Below I use the melt function from the data.table package to reformat the data into stacked bar graph format. 
library(data.table)
library(ggplot2)
library(ggpubr)
library(RColorBrewer)
leaves_phy_melt <- melt(leaves_phylum_rel_abund, id.vars = 'Treatment', variable.name = 'Phylum')
leaves_colors_n_phylum <- length(unique(leaves_phy_melt[, 'Phylum']))
leaves_phy_melt$Treatment <- as.character(leaves_phy_melt$Treatment)
leaves_phy_melt$Treatment <- factor(leaves_phy_melt$Treatment, levels = c("Control 2018", "Control 2019", "Pre-burned 2018", "Post-burned 2019"))

tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Leaves_16s/Results/leaves_16s_relative_abundance_phylum.tiff", units = 'in', width = 7, height = 7, res = 300)
leaves_phylum_rel_abund_chart <- ggplot(leaves_phy_melt, aes(x = Treatment, y = value, fill = Phylum))+
  geom_bar(stat = 'identity', show.legend = TRUE, color = 'black')+
  scale_fill_manual(values = colorRampPalette(brewer.pal(7, 'Accent'))(leaves_colors_n_phylum))+
  xlab('Leaves')+
  ylab('Relative Abundance')+
  ggtitle('Leaves - Phylum level')+
  theme(panel.border = element_blank(),
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x = element_text(size = 11, color = 'black', angle = 90),
        axis.ticks.x = element_blank(), 
        axis.text.y = element_text(size = 14), 
        axis.line.y.left = element_line(), 
        axis.title.y = element_blank(), 
        legend.title = element_text(face = "bold", size = 14),
        legend.text = element_text(size = 14))
leaves_phylum_rel_abund_chart
dev.off()

leaves_phylum_percent <- leaves_phylum_rel_abund * 100 
leaves_phylum_percent <- leaves_phylum_percent[, -1]
leaves_phylum_percent <- data.frame(Treatment = leaves_phylum_rel_abund$Treatment, leaves_phylum_percent)
is.num <- sapply(leaves_phylum_percent, is.numeric)
leaves_phylum_percent[is.num] <- lapply(leaves_phylum_percent[is.num], round, 2)
leaves_proteobac_mean <- mean(leaves_phylum_percent$Proteobacteria)
leaves_proteobac_mean
```

# The following chunk creates a relative abundance chart at the class level.

```{r Relative abundance chart - Class}
library(plyr)
leaves_otu_class <- data.frame(class = leaves_otu_taxonomy$class, trans_otu_leaves)

#I then remove the assignment confidence values contained in parentheses found in each taxonomic string using the sub function.
leaves_otu_class$class <- sub("\\(.*)","",x = leaves_otu_class$class)
leaves_summed_otu_class <- as.data.frame(ddply(leaves_otu_class, .(class), colwise(sum)))
rownames(leaves_summed_otu_class) <- make.names(leaves_summed_otu_class$class)
leaves_summed_otu_class <- leaves_summed_otu_class[, 2:41]

#I then transpose the dataframe and merge the metadata information
trans_leaves_class <- as.data.frame(t(leaves_summed_otu_class))
rownames(trans_leaves_class) <- sub('X', '', x = rownames(trans_leaves_class))
trans_leaves_class_metadata <- merge(leaves_metadata_for_rel_abund, trans_leaves_class, by.x = 1, by.y = 0)

#Then we can delete the column with the sample names as they are not needed and then sum by treatment
trans_leaves_class_metadata <- trans_leaves_class_metadata[, 2:35]
leaves_class_summed_treatment <- as.data.frame(ddply(trans_leaves_class_metadata, .(Treatment), colwise(sum)))

#I then subset the data so that only numerical data is present and creating a new class of objects, other, that includes OTUs from classes that represent 2% or less of the total community.
leaves_class <- leaves_class_summed_treatment[, 2:34]
leaves_class_others <- leaves_class[, colSums(leaves_class)/sum(leaves_class) <= 0.01]
leaves_class_others_sum <- rowSums(leaves_class_others)
leaves_class_regular <- leaves_class[, colSums(leaves_class)/sum(leaves_class) > 0.01]

#I then create a new dataframe that contains all of the class data and determine relative abundace using decostand function from vegan
leaves_class_reg_others <- data.frame(Treatment = leaves_class_summed_treatment$Treatment, leaves_class_regular, Others = leaves_class_others_sum)
library(vegan)
leaves_class_rel_abund <- decostand(leaves_class_reg_others[, 2:11], method = 'total')
leaves_class_rel_abund <- data.frame(Treatment = leaves_class_reg_others$Treatment, leaves_class_rel_abund)

#Below I use the melt function from the data.table package to reformat the data into stacked bar graph format.
library(data.table)
library(ggplot2)
library(ggpubr)
library(RColorBrewer)
leaves_class_melt <- melt(leaves_class_rel_abund, id.vars = 'Treatment', variable.name = 'Class')
leaves_class_n_color <- length(unique(leaves_class_melt$Class))
leaves_class_melt$Treatment <- as.character(leaves_class_melt$Treatment)
leaves_class_melt$Treatment <- factor(leaves_class_melt$Treatment, levels = c("Control 2018", "Control 2019", "Pre-burned 2018", "Post-burned 2019"))

tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Leaves_16s/Results/leaves_16s_relative_abundance_class.tiff", units = 'in', width = 7, height = 7, res = 300)
leaves_class_rel_abund_chart <- ggplot(leaves_class_melt, aes(x = Treatment, y = value, fill = Class))+
  geom_bar(stat = 'identity', show.legend = TRUE, color = 'black')+
  scale_fill_manual(values = colorRampPalette(brewer.pal(7, 'Accent'))(leaves_class_n_color))+
  xlab('Leaves')+
  ylab('Relative Abundance')+
  ggtitle('Leaves - Class level')+
  theme(panel.border = element_blank(), 
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.title.x = element_blank(), 
        axis.text.x = element_text(size = 12, color = 'black', angle = 90), 
        axis.ticks.x = element_blank(), 
        axis.text.y = element_text(size = 12), 
        axis.line.y.left = element_line(), 
        axis.title.y = element_blank())
dev.off()

#Let's join the relative abundance charts at phylum and class level
tiff(filename = "/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Leaves_16s/Results/16s_leaves_rel_abund_phylum_class.tiff", units = 'in', width = 10, height = 7, res = 200)
ggarrange(leaves_phylum_rel_abund_chart, leaves_class_rel_abund_chart,  nrow = 1, labels = c("I", "II"), align = 'h')
dev.off()

leaves_class_percent <- leaves_class_rel_abund * 100 
leaves_class_percent <- leaves_class_percent[, -1]
leaves_class_percent <- data.frame(Treatment = leaves_class_rel_abund$Treatment, leaves_class_percent)
is.num <- sapply(leaves_class_percent, is.numeric)
leaves_class_percent[is.num] <- lapply(leaves_class_percent[is.num], round, 2)
leaves_alphaproteobac_mean <- mean(leaves_class_percent$Alphaproteobacteria)
leaves_alphaproteobac_mean
```

# The following chunk creates the relative abundance chart for leaves 16s niche at order level

```{r Relative abundance Chart - Order}
leaves_otu_order <- data.frame(order = leaves_otu_taxonomy$order, trans_otu_leaves)

#I then remove the assignment confidence values contained in parentheses found in each taxonomic string using the sub function.
leaves_otu_order$order <- sub("\\(.*)","",x = leaves_otu_order$order)
leaves_summed_order <- as.data.frame(ddply(leaves_otu_order, .(order), colwise(sum)))
rownames(leaves_summed_order) <- make.names(leaves_summed_order$order)
leaves_summed_order <- leaves_summed_order[, 2:41]

#I then transpose the dataframe and merge the metadata information
trans_leaves_order <- as.data.frame(t(leaves_summed_order))
rownames(trans_leaves_order) <- sub('X', '', rownames(trans_leaves_order))
trans_leaves_order_metadata <- merge(leaves_metadata_for_rel_abund, trans_leaves_order, by.x = 1, by.y = 0)

#Then we can delete the column with the sample names as they are not needed and then sum by treatment
trans_leaves_order_metadata <- trans_leaves_order_metadata[, 2:80]
leaves_order_summed_treatment <- as.data.frame(ddply(trans_leaves_order_metadata, .(Treatment), colwise(sum)))

#I then subset the data so that only numerical data is present and creating a new class of objects, other, that includes OTUs from orders that represent 2% or less of the total community.
leaves_order <- leaves_order_summed_treatment[, 2:79]
leaves_order_others <- leaves_order[, colSums(leaves_order)/sum(leaves_order) <= 0.02]
leaves_order_others_sum <- rowSums(leaves_order_others)
leaves_order_regular <- leaves_order[, colSums(leaves_order)/sum(leaves_order) > 0.02]

#I then create a new dataframe that contains all of the class data and determine relative abundace using decostand function from vegan
leaves_order_reg_others <- data.frame(Treatment = leaves_order_summed_treatment$Treatment, leaves_order_regular, Others = leaves_order_others_sum)
library(vegan)
leaves_order_rel_abund <- decostand(leaves_order_reg_others[, 2:10], method = 'total')
leaves_order_rel_abund <- data.frame(Treatment = leaves_order_reg_others$Treatment, leaves_order_rel_abund)

#Below I use the melt function from the data.table package to reformat the data into stacked bar graph format.
library(data.table)
library(ggplot2)
library(ggpubr)
library(RColorBrewer)
leaves_order_melt <- melt(leaves_order_rel_abund, id.vars = 'Treatment', variable.name = 'order')
leaves_order_n_color <- length(unique(leaves_order_melt$order))
leaves_order_melt$Treatment <- as.character(leaves_order_melt$Treatment)
leaves_order_melt$Treatment <- factor(leaves_order_melt$Treatment, levels = c('Pre_control_A', 'Post_control_A', 'Pre_control_D', 'Post_control_D', 'Pre_burn_B', 'Post_burn_B', 'Pre_burn_C', 'Post_burn_C'))

tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Leaves_16s/Results/leaves_16s_relative_abundance_order.tiff", units = 'in', width = 7, height = 7, res = 300)
leaves_order_rel_abund_chart <- ggplot(leaves_order_melt, aes(x = Treatment, y = value, fill = order))+
  geom_bar(stat = 'identity', show.legend = TRUE, color = 'black')+
  scale_fill_manual(values = colorRampPalette(brewer.pal(7, 'Accent'))(leaves_order_n_color))+
  xlab('Leaves')+
  ylab('Relative Abundance')+
  theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.x = element_text(size = 16), axis.text.x = element_text(size = 14, color = 'black', angle = 90), axis.ticks.x = element_blank(), axis.text.y = element_text(size = 14), axis.line.y.left = element_line(), axis.title.y = element_text(size = 16))
dev.off()
```

# The purpose of the following chunk is to create Shannon Alpha diversity boxplots across treatments

```{r}
library(ggpubr)
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Leaves_16s/Results/leaves_relative_abundance_phylum_class_order.tiff", units = 'in', width = 12, height = 7, res = 200)
ggarrange(leaves_phylum_rel_abund_chart, leaves_class_rel_abund_chart, leaves_order_rel_abund_chart, ncol=3,nrow=1,labels=c("A","B","C"))
dev.off()
```

```{r Alpha Diversity}
#The following function from vegan will give us the shannon diversity index
leaves_alpha_diversity <- diversity(rare_otu_leaves, index = 'shannon')
leaves_alpha_diversity <- as.data.frame(leaves_alpha_diversity)

#Adding in the metadata for graph
leaves_alpha_metadata <- merge(leaves_metadata_for_rel_abund, leaves_alpha_diversity, by.x = 1, by.y = 0)
rownames(leaves_alpha_metadata) <- make.names(leaves_alpha_metadata$Sample)
leaves_alpha_metadata <- leaves_alpha_metadata[, 2:3]
rownames(leaves_alpha_metadata) <- sub('X', '', rownames(leaves_alpha_metadata))
leaves_alpha_metadata$Treatment <- as.character(leaves_alpha_metadata$Treatment)
leaves_alpha_metadata$Treatment <- factor(leaves_alpha_metadata$Treatment, levels = c('Pre_control_A', 'Post_control_A', 'Pre_control_D', 'Post_control_D', 'Pre_burn_B', 'Post_burn_B', 'Pre_burn_C', 'Post_burn_C'))

#Let's make the graph now
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/leaves_16s/Results/16s_leaves_alpha_shannon_index.tiff", units = 'in', width = 10, height = 7, res = 200)
ggplot(leaves_alpha_metadata, aes(x = Treatment, y = leaves_alpha_diversity, fill = Treatment))+
  geom_boxplot(outlier.shape = NA)+
  ggtitle('Leaves 16S Alpha Diversity')+
  ylab('Shannon Diversity Index')+
  xlab('Treatment')+
  theme_linedraw()+
  theme(axis.title.x = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(plot.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.x = element_text(family = 'Times New Roman', size = 14, colour = 'black', face = 'bold', angle = 90))+
  theme(axis.title.y = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.y = element_text(family = 'Times New Roman', size = 14, face = 'bold'))+
  theme(legend.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(legend.text = element_text(size = 18, family = 'Times New Roman'))+
  scale_fill_brewer(palette = 'Set1')+
  guides(fill=guide_legend(title = 'Treatment'))
dev.off()

#Let's make the graph now
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/leaves_16s/Results/16s_leaves_alpha_shannon_index_without_legend.tiff", units = 'in', width = 10, height = 7, res = 200)
leaves_alpha_without_legend <- ggplot(leaves_alpha_metadata, aes(x = Treatment, y = leaves_alpha_diversity, fill = Treatment))+
  geom_boxplot(outlier.shape = NA)+
  ggtitle('Leaves 16S Alpha Diversity')+
  ylab('Shannon Index')+
  xlab('Treatment')+
  theme_linedraw()+
  theme(axis.title.x = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(plot.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.x = element_text(family = 'Times New Roman', size = 14, colour = 'black', face = 'bold', angle = 90))+
  theme(axis.title.y = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.y = element_text(family = 'Times New Roman', size = 14, face = 'bold'))+
  theme(legend.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(legend.text = element_text(size = 18, family = 'Times New Roman'))+
  theme(legend.position = 'none')+
  scale_fill_brewer(palette = 'Set1')+
  guides(fill=guide_legend(title = 'Treatment'))
dev.off()
```

# The following code is used to generate the Principal Coordinate Analysis chart

```{r Principal Coordinate Analyses}
#We will start with the object named 'rare_otu_leaves_no_singletons'
library(vegan)
rare_leaves_rel_abund_for_pcoa <- decostand(rare_otu_leaves_no_singletons, method = 'total')

#Need to load the library 'ape' and use the vegdist function to create a dissimilarity matrix using the Bray Curtis distance
library(ape)
leaves_dist <- vegdist(rare_leaves_rel_abund_for_pcoa, method = 'bray')
leaves_pcoa <- pcoa(leaves_dist)
leaves_pcoa_vectors <- as.data.frame(leaves_pcoa$vectors)
leaves_pcoa_vectors_metadata <- merge(leaves_metadata_for_rel_abund, leaves_pcoa_vectors, by.x = 1, by.y = 0)
biplot(leaves_pcoa, plot.axes = c(1, 2))
leaves_pcoa_scores <- data.frame(PC1 = leaves_pcoa_vectors$Axis.1, PC2 = leaves_pcoa_vectors$Axis.2)
leaves_pcoa_ellipse <- ordiellipse(leaves_pcoa_scores, leaves_pcoa_vectors_metadata$Treatment)

#Plot the eigenvalues and interpret
barplot(leaves_pcoa$values$Relative_eig[1:10])

# Find out the percentage variance explained by the first two principal axes
leaves_pcoa$values$Relative_eig[1] * 100 #21.71
leaves_pcoa$values$Relative_eig[2] * 100 #14.74

#Extract the plot scores from first two PCoA axes
leaves_pcoa_axes <- leaves_pcoa$vectors[, c(1, 2)]

#In order to change the color scheme of PCoA we will have to add one more column to the dataframe 'leaves_pcoa_vectors_metadata'
leaves_treatment_pcoa_color <- c('Burn_B', 'Burn_B','Burn_B','Burn_B','Burn_B', 'Burn_C', 'Burn_C','Burn_C','Burn_C','Burn_C', 'Control_D', 'Control_D','Control_D','Control_D','Control_D', 'Control_A', 'Control_A','Control_A','Control_A','Control_A', 'Burn_B', 'Burn_B','Burn_B','Burn_B','Burn_B', 'Burn_C', 'Burn_C','Burn_C','Burn_C','Burn_C', 'Control_D', 'Control_D','Control_D','Control_D','Control_D', 'Control_A', 'Control_A','Control_A','Control_A','Control_A')
leaves_pcoa_vectors_metadata <- data.frame(Treatment_color = leaves_treatment_pcoa_color, leaves_pcoa_vectors_metadata)

tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/leaves_16s/Results/16s_leaves_PCOA.tiff", units = 'in', width = 10, height = 7, res = 200)
leaves_pcoa_16s <- ggplot(leaves_pcoa_scores, aes(PC1, PC2, color = leaves_pcoa_vectors_metadata$Treatment_color))+
  geom_point(size = 2, aes(shape = year_permanova_leaves))+
  labs(shape = 'Year')+
  ggtitle('Leaves 16S')+
  xlab('Principal Axis 1 (21.71%)')+
  xlim(c(-0.6, 0.6))+
  ylab('Principal Axis 2 (14.74%)')+
  theme_linedraw()+
  #scale_color_discrete(name = 'Niches')+
  theme(plot.title = element_text(family = 'Arial', size=14, face="bold"))+
  theme(axis.title.x=element_text(family = 'Arial', size=10))+
  theme(axis.title.y=element_text(family = 'Arial', size=10))+
  theme(axis.text.x=element_text(family = 'Arial', size=10))+
  theme(axis.text.y=element_text(family = 'Arial', size=10))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 14))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))+
  scale_color_brewer(palette = 'Set1', name = 'Treatment')
dev.off()
```

# The purpose of this code is to PERMANOVA on the WHOLE data

```{r}
# PERMANOVA without plot information
treatment_leaves_whole <- c("Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control")

year_permanova_leaves <- c("2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019")

leaves_permanova_whole <- adonis(rare_leaves_rel_abund_for_pcoa~treatment_leaves_whole + year_permanova_leaves + treatment_leaves_whole * year_permanova_leaves, method = 'bray', permutations = 10000)
leaves_permanova_whole

leaves_plot <- c("B","B","B","B","B","C","C","C","C","C","D","D","D","D","D","A","A","A","A","A","B","B","B","B","B","C","C","C","C","C","D","D","D","D","D","A","A","A","A","A")

#PERMANOVA with plot information
leaves_permanova_whole_plot <- adonis(rare_leaves_rel_abund_for_pcoa ~ treatment_leaves_whole +  year_permanova_leaves + leaves_plot + treatment_leaves_whole * year_permanova_leaves * leaves_plot, method = 'bray', permutations = 10000)
leaves_permanova_whole_plot
```

#The purpose of this code chunk is to find differences in shannon alpha diversity across treatments.

```{r Differences in shannon index}
#We have to generate a new object for plot information
plot_tree_leaves <- c("B_1", "B_2","B_3","B_4","B_5", "C_1", "C_2","C_3","C_4","C_5", "D_1", "D_2", "D_3", "D_4", "D_5", "A_1", "A_2","A_3","A_4","A_5","B_1", "B_2","B_3","B_4","B_5", "C_1", "C_2","C_3","C_4","C_5", "D_1", "D_2", "D_3", "D_4", "D_5", "A_1", "A_2","A_3","A_4","A_5")

#We will start with object 'leaves_alpha_diversity_treatment' and add one more column to it and that is plot information
leaves_plot_shannon <- data.frame(Plot = plot_tree_leaves, Year = year_permanova_leaves, Treatment = treatment_leaves_whole, leaves_alpha_metadata)
leaves_plot_shannon <- leaves_plot_shannon[, -4]
colnames(leaves_plot_shannon)[4] <- "Shannon_index"

#Install required packages if they are not already installed
if(!require(psych)){install.packages("psych")}
if(!require(nlme)){install.packages("nlme")}
if(!require(car)){install.packages("car")}
if(!require(multcompView)){install.packages("multcompView")}
if(!require(lsmeans)){install.packages("lsmeans")}
if(!require(ggplot2)){install.packages("ggplot2")}
if(!require(rcompanion)){install.packages("rcompanion")}

#Order factors by order in dataframe, otherwise, R will alphabetize them
leaves_plot_shannon$Treatment <- factor(leaves_plot_shannon$Treatment, levels = unique(leaves_plot_shannon$Treatment))

#Convert the Year column to numeric
leaves_plot_shannon$Year <- as.numeric(leaves_plot_shannon$Year)

#Code for the analysis
library(nlme)
leaves_model_alpha_diversity <- lme(Shannon_index ~ Treatment + Year + Treatment*Year, random = ~1|Plot, correlation = corAR1(form = ~ Year | Plot, value = 0.4287), data = leaves_plot_shannon, method = "REML")
library(car)
Anova(leaves_model_alpha_diversity)

#Code for the post hoc analysis
leaves_plot_shannon$treatment_year <- interaction(leaves_plot_shannon$Treatment, leaves_plot_shannon$Year)
leaves_alpha_model <- lme(Shannon_index ~ treatment_year, random = ~1|Plot, correlation = corAR1(form = ~ Year | Plot, value = 0.4287), data = leaves_plot_shannon, method = "REML")

#Let's perform the tukey test on leaves alpha diversity
library(lsmeans)
leaves_alpha_post_hoc <- lsmeans(leaves_alpha_model, ~treatment_year, adjust = "tukey")
cld(leaves_alpha_post_hoc, alpha = 0.05, Letters = letters, adjust = "tukey")
```

#The purpose of this code is find the differences in observed richness across treatments.

```{r Differences in Richness}
# RICHNESS
library(vegan)
leaves_richness <- specnumber(rare_otu_leaves)
leaves_richness_metadata <- data.frame(Plot = plot_tree_leaves, Year = year_permanova_leaves, Treatment = treatment_leaves_whole, Richness = leaves_richness)

#Order factors by order in dataframe, otherwise, R will alphabetize them
leaves_richness_metadata$Treatment <- factor(leaves_richness_metadata$Treatment, levels = unique(leaves_richness_metadata$Treatment))

#Make the Year column numeric
leaves_richness_metadata$Year <- as.numeric(leaves_richness_metadata$Year)

#Code for the analysis
library(nlme)
leaves_model_richness <- lme(Richness ~ Treatment + Year + Treatment*Year, random = ~1|Plot, correlation = corAR1(form = ~ Year | Plot, value = 0.4287), data = leaves_richness_metadata, method = "REML")
library(car)
Anova(leaves_model_richness)

#Code for the post hoc analysis
leaves_richness_metadata$treatment_year <- interaction(leaves_richness_metadata$Treatment, leaves_richness_metadata$Year)
leaves_richness_model_tukey <- lme(Richness ~ treatment_year, random = ~1|Plot, correlation = corAR1(form = ~ Year | Plot, value = 0.4287), data = leaves_richness_metadata, method = "REML")

#Let's perform the tukey test on leaves alpha diversity
library(lsmeans)
leaves_richness_post_hoc <- lsmeans(leaves_alpha_model, ~treatment_year, adjust = "tukey")
cld(leaves_richness_post_hoc, alpha = 0.05, Letters = letters, adjust = "tukey")

#One thing that we forgot is to make charts for the richness. We will do that now. Let's create the data frame required.
leaves_richness_df <- data.frame(leaves_alpha_metadata, Richness = leaves_richness)
leaves_richness_df <- leaves_richness_df[, -2]
leaves_richness_df$Treatment <- as.character(leaves_richness_df$Treatment)
leaves_richness_df$Treatment <- factor(leaves_richness_df$Treatment, levels = c('Pre_control_A', 'Post_control_A', 'Pre_control_D', 'Post_control_D', 'Pre_burn_B', 'Post_burn_B', 'Pre_burn_C', 'Post_burn_C'))

# This is the graph with legend
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/leaves_16s/Results/16s_leaves_richness.tiff", units = 'in', width = 10, height = 7, res = 200)
ggplot(leaves_richness_df, aes(x = Treatment, y = Richness, fill = Treatment))+
  geom_boxplot(outlier.shape = NA)+
  ggtitle('Leaves 16S Richness')+
  ylab('OTU Richness')+
  xlab('Treatment')+
  theme_linedraw()+
  theme(axis.title.x = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(plot.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.x = element_text(family = 'Times New Roman', size = 14, colour = 'black', face = 'bold', angle = 90))+
  theme(axis.title.y = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.y = element_text(family = 'Times New Roman', size = 14, face = 'bold'))+
  theme(legend.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(legend.text = element_text(size = 18, family = 'Times New Roman'))+
  scale_fill_brewer(palette = 'Set1')+
  guides(fill=guide_legend(title = 'Treatment'))
dev.off()

#Let's make the graph without legend
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/leaves_16s/Results/16s_leaves_richness_without_legend.tiff", units = 'in', width = 8, height = 7, res = 200)
leaves_richness_chart_without_legend <- ggplot(leaves_richness_df, aes(x = Treatment, y = Richness, fill = Treatment))+
  geom_boxplot(outlier.shape = NA)+
  ggtitle('Leaves 16S Richness')+
  ylab('OTU Richness')+
  xlab('Treatment')+
  theme_linedraw()+
  theme(axis.title.x = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(plot.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.x = element_text(family = 'Times New Roman', size = 14, colour = 'black', face = 'bold', angle = 90))+
  theme(axis.title.y = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.y = element_text(family = 'Times New Roman', size = 14, face = 'bold'))+
  theme(legend.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(legend.text = element_text(size = 18, family = 'Times New Roman'))+
  theme(legend.position = 'none')+
  scale_fill_brewer(palette = 'Set1')+
  guides(fill=guide_legend(title = 'Treatment'))
dev.off()

#Now let's combine the alpha diversity and richness chart using ggarrange
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/leaves_16s/Results/16s_leaves_richness_and_shannon.tiff", units = 'in', height = 10, width = 8, res = 300)
ggarrange(leaves_richness_chart_without_legend + theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()), leaves_alpha_without_legend,  ncol = 1, labels = c("I", "II"), align = 'v', hjust = -3)
dev.off()
```

```{r Concise box plots}
#The purpose of this code chunk is to make the shannon diversity box plots in the concise form. We will start working with dataframe leaves_alpha_diversity
leaves_16s_alpha_diversity <- leaves_alpha_diversity
colnames(leaves_16s_alpha_diversity)[1] <- 'values'

#Note that the metadata that we wanna use now has already been created for pcoa plot so will use that metadata.
leaves_treatment_pcoa_color
year_permanova_leaves
leaves_16s_alpha_concise <- data.frame(leaves_16s_alpha_diversity, Treatment = leaves_treatment_pcoa_color, Year = year_permanova_leaves)
leaves_16s_alpha_concise$Treatment <- as.character(leaves_16s_alpha_concise$Treatment)
leaves_16s_alpha_concise$Treatment <- factor(leaves_16s_alpha_concise$Treatment, levels = c('Control_A', 'Control_D', 'Burn_B', 'Burn_C'))

leaves_16s_alpha <- ggplot(data = leaves_16s_alpha_concise, aes(x = Treatment, y = values)) + geom_boxplot(aes(fill = Year)) +
  ggtitle('Leaves 16S')+
  xlab('Treatment')+
  ylab('Shannon Diversity Index')+
  theme_linedraw()+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  theme(axis.text.x=element_text(family = 'Arial', size=12))+
  theme(axis.text.y=element_text(family = 'Arial', size=12))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))+
  scale_color_brewer(palette = 'Set1', name = 'Treatment')

#We have to do the same for the richness graph. We will use leaves_richness_df dataframe
leaves_16s_richness <- leaves_richness_df

#We wanna delete the existing Treatment column
leaves_16s_richness <- subset(leaves_16s_richness, select = -Treatment)

#Add the same metadata previously added
leaves_16s_richness <- data.frame(leaves_16s_richness, Treatment = leaves_treatment_pcoa_color, Year = year_permanova_leaves)
leaves_16s_richness$Treatment <- as.character(leaves_16s_richness$Treatment)
leaves_16s_richness$Treatment <- factor(leaves_16s_richness$Treatment, levels = c('Control_A', 'Control_D', 'Burn_B', 'Burn_C'))

leaves_16s_richness_graph <- ggplot(data = leaves_16s_richness, aes(x = Treatment, y = Richness)) + geom_boxplot(aes(fill = Year)) +
  #ggtitle('Leaves 16S')+
  xlab('Treatment')+
  ylab('OTU Richness')+
  theme_linedraw()+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  scale_y_continuous(position = "right")+
  theme(axis.text.x=element_text(family = 'Arial', size=12))+
  theme(axis.text.y=element_text(family = 'Arial', size=12))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))+
  scale_color_brewer(palette = 'Set1', name = 'Treatment')
```

```{r Anova test on 2018 data}
# We will start with subsetting the leaves_alpha_diversity_treatment dataframe
leaves_alpha_2018 <- as.data.frame(leaves_alpha_diversity[1:20, ])
colnames(leaves_alpha_2018)[1] <- 'alpha_diversity'

plot_leaves_anova <- c("B", "B","B","B","B","C","C","C","C","C","D","D","D","D","D","A","A","A","A","A")
leaves_alpha_2018 <- data.frame(Plot = plot_leaves_anova, leaves_alpha_2018)

leaves_alpha_2018_anova <- aov(alpha_diversity ~ Plot, data = leaves_alpha_2018)
summary(leaves_alpha_2018_anova)
```

#The purpose of this code chunk is to prepare alpha diversity and richness boxplots without plot information

```{r Box plots without plot}
#The purpose of this code chunk is to make the shannon diversity box plots in the concise form. We will start working with dataframe leaves_alpha_diversity
leaves_16s_alpha_diversity <- leaves_alpha_diversity
colnames(leaves_16s_alpha_diversity)[1] <- 'values'

#Create metadata
leaves_treatment <- c("Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Control", "Control","Control","Control","Control","Control","Control","Control","Control","Control","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Control", "Control","Control","Control","Control","Control","Control","Control","Control","Control")
year_permanova_leaves
leaves_16s_alpha_without_plot <- data.frame(leaves_16s_alpha_diversity, Treatment = leaves_treatment, Year = year_permanova_leaves)
leaves_16s_alpha_without_plot$Treatment <- as.character(leaves_16s_alpha_without_plot$Treatment)
leaves_16s_alpha_without_plot$Treatment <- factor(leaves_16s_alpha_without_plot$Treatment, levels = c("Control", "Burned"))

leaves_16s_alpha_graph_without_plot <- ggplot(data = leaves_16s_alpha_without_plot, aes(x = Treatment, y = values)) + geom_boxplot(aes(fill = Year), outlier.shape = NA) +
  ggtitle('Leaves 16S')+
  xlab('Treatment')+
  ylab('Shannon Diversity Index')+
  theme_linedraw()+
  geom_text(data=NULL,aes(x=0.81, y=5.05, label="A"))+
  geom_text(data=NULL,aes(x=1.19, y=5.0, label="A"))+ 
  geom_text(data=NULL,aes(x=1.81, y=4.85, label="A"))+ 
  geom_text(data=NULL,aes(x=2.19, y=4.75, label="A"))+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  theme(axis.text.x=element_text(family = 'Arial', size=12))+
  theme(axis.text.y=element_text(family = 'Arial', size=12))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))+
  scale_color_brewer(palette = 'Set1', name = 'Treatment')

leaves_16s_alpha_graph_without_plot

#We have to do the same for the richness graph. We will use leaves_richness_df dataframe
leaves_16s_richness <- leaves_richness_df

#We wanna delete the existing Treatment column
leaves_16s_richness <- subset(leaves_16s_richness, select = -Treatment)

#Add the same metadata previously added
leaves_16s_richness_without_plot <- data.frame(leaves_16s_richness, Treatment = leaves_treatment, Year = year_permanova_leaves)
leaves_16s_richness_without_plot$Treatment <- as.character(leaves_16s_richness_without_plot$Treatment)
leaves_16s_richness_without_plot$Treatment <- factor(leaves_16s_richness_without_plot$Treatment, levels = c("Control", "Burned"))

leaves_16s_richness_graph_without_plot <- ggplot(data = leaves_16s_richness_without_plot, aes(x = Treatment, y = Richness)) + geom_boxplot(aes(fill = Year), outlier.shape = NA) +
  #ggtitle('leaves 16S')+
  xlab('Treatment')+
  ylab('OTU Richness')+
  theme_linedraw()+
  geom_text(data=NULL,aes(x=0.81, y=340, label="A"))+
  geom_text(data=NULL,aes(x=1.19, y=490, label="A"))+ 
  geom_text(data=NULL,aes(x=1.81, y=300, label="A"))+ 
  geom_text(data=NULL,aes(x=2.19, y=330, label="A"))+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  scale_y_continuous(position = "right")+
  theme(axis.text.x=element_text(family = 'Arial', size=12))+
  theme(axis.text.y=element_text(family = 'Arial', size=12))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))+
  scale_color_brewer(palette = 'Set1', name = 'Treatment')

leaves_16s_richness_graph_without_plot
```

#The purpose of this chunk is to perform the indicator species analysis on leaves data

```{r Indicator Species Analysis}
treatment_leaves_indicator <- c("Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019")

#We are pretty much ready to run the analysis on leaves data
leaves_indval <- multipatt(rare_otu_leaves_no_singletons, treatment_leaves_indicator, duleg = TRUE, func = "IndVal.g", control = how(nperm = 10000))
sink(file = "/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Leaves_16s/Tables_generated_from_R/leaves_16s_indicator_species.txt")
summary(leaves_indval, indvalcomp = TRUE)
sink()

library(readxl)
leaves_indicators <- read_excel("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Leaves_16s/Tables_from_R/16s_leaves_indicators.xlsx", col_names = TRUE)
leaves_indicators <- as.data.frame(leaves_indicators)
rownames(leaves_indicators) <- leaves_indicators[, 2]
leaves_indicators_test <- leaves_indicators[, -2]

#Subset the dataframes based on treatment
leaves_indic_control_2018 <- subset(leaves_indicators_test, leaves_indicators_test$Treatment == 'Control 2018')
leaves_indic_control_2019 <- subset(leaves_indicators_test, leaves_indicators_test$Treatment == 'Control 2019')
leaves_indic_burned_2018 <- subset(leaves_indicators_test, leaves_indicators_test$Treatment == 'Burned 2018')
leaves_indic_burned_2019 <- subset(leaves_indicators_test, leaves_indicators_test$Treatment == 'Burned 2019')

#Keep first ten rows
leaves_indic_control_2018_imp <- leaves_indic_control_2018[1:10, ]
leaves_indic_control_2019_imp <- leaves_indic_control_2019[1:10, ]
leaves_indic_burned_2018_imp <- leaves_indic_burned_2018[1:10, ]
leaves_indic_burned_2019_imp <- leaves_indic_burned_2019[1:10, ]

leaves_indic_imp <- rbind(leaves_indic_control_2018_imp, leaves_indic_control_2019_imp, leaves_indic_burned_2018_imp, leaves_indic_burned_2019_imp)

#Create one more dataframe only with OTU and treatment info
leaves_indic_treatment <- as.data.frame(leaves_indic_imp[, 1])
colnames(leaves_indic_treatment) <- "Treatment"
rownames(leaves_indic_treatment) <- rownames(leaves_indic_imp)

leaves_indic_treatment$Treatment <- factor(leaves_indic_treatment$Treatment, levels = c("Control 2018", "Control 2019", "Burned 2018", "Burned 2019"))

library(RColorBrewer)
library(grid)
leaves_ind_heatmap <- pheatmap(leaves_indic_imp[, 2:4], 
                              treeheight_row = 0, 
                              treeheight_col = 0,
                              annotation_row = leaves_indic_treatment,
                              color = colorRampPalette(brewer.pal(8, "Blues"))(100),
                              cluster_rows = F,
                              gaps_row = c(10, 20, 30))
setHook("grid.newpage", function() pushViewport(viewport(x=1,y=1,width=0.9, height=0.9, name="vp", just=c("right","top"))), action="prepend")
leaves_ind_heatmap
setHook("grid.newpage", NULL, "replace")
grid.text("xlabel example", y=-0.07, gp=gpar(fontsize=16))
grid.text("16S leaves", y = -0.07, gp = gpar(fontsize = 16))
dev.off()

#To find out the identity of the indicator species we have to merge indicator species dataframe to leaves taxonomy dataframe
library(tidyr)
leaves_indic_taxonomy <- merge(leaves_indic_imp, leaves_otu_taxonomy, by = 0)

#Subset by treatment so that we get OTUs which belong to "Burned 2019"
leaves_indic_tax_burned_2019 <- leaves_indic_taxonomy[leaves_indic_taxonomy$Treatment == "Burned 2019", ]
leaves_indic_tax_burned_2019 <- leaves_indic_tax_burned_2019[order( - leaves_indic_tax_burned_2019$Indicator_value), ]
```

# This is hopefully the last way I am making my boxplots from richness and alpha diversity. Let's start from a fresh dataset

```{r}
# Use leaves_16s_alpha_without_plot and make a new dataframe out of it
alpha_leaves <- leaves_16s_alpha_without_plot

# Prepare a new vector
alpha_leaves_treatment <- c("Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019", "Control 2019", "Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019")

#Add the above vector to alpha_leaves
alpha_leaves <- data.frame(alpha_treatment = alpha_leaves_treatment, alpha_leaves)

#Delete Treatment and year column
alpha_leaves_clean <- alpha_leaves[, -c(3, 4)]

#Change the column name alpha_treatment to simply Treatment
colnames(alpha_leaves_clean)[1] <- "Treatment"

#Convert the variable Treatment to factor variable
alpha_leaves_clean$Treatment <- factor(alpha_leaves_clean$Treatment, levels = c("Control 2018", "Control 2019", "Pre-burned 2018", "Post-burned 2019"))

#Let's make the boxplot
library(ggplot2)
p_leaves <- ggplot(alpha_leaves_clean, aes(x = Treatment, y = values, fill = Treatment)) +
  geom_boxplot(outlier.shape = NA)+
  theme_linedraw()+
  scale_fill_manual(values = alpha(c("lightgreen","springgreen4","bisque1","firebrick3"), 0.9))+
  ggtitle('Leaves 16S')+
  #xlab('Treatment')+
  #ylab('Shannon Diversity Index')+
  theme(legend.title = element_text(face = "bold", size = 14),
        legend.text = element_text(size = 14))+
  theme(plot.title = element_text(face = "bold", size = 14))+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  theme(axis.text.x=element_text(family = 'Arial', size=10, angle = 10, vjust = 0.6))

p_leaves

#Have to do the same for richness. Let's start with the dataframe leaves_16s_richness and add Treatment column
leaves_rich <- data.frame(leaves_16s_richness, Treatment = alpha_leaves_treatment)
leaves_rich$Treatment <- factor(leaves_rich$Treatment, levels = c("Control 2018", "Control 2019", "Pre-burned 2018", "Post-burned 2019"))

#Let's make the boxplot
library(ggplot2)
r_leaves <- ggplot(leaves_rich, aes(x = Treatment, y = Richness, fill = Treatment)) +
  geom_boxplot(outlier.shape = NA)+
  theme_linedraw()+
  scale_fill_manual(values = alpha(c("lightgreen","springgreen4","bisque1","firebrick3"), 0.9))+
  theme(legend.title = element_text(face = "bold", size = 14),
        legend.text = element_text(size = 14))+
  theme(plot.title = element_text(face = "bold", size = 14))+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  scale_y_continuous(position = "right")+
  theme(axis.text.x=element_text(family = 'Arial', size=10, angle = 10, vjust = 0.6))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))

r_leaves
```

#T.test by treatment

```{r}
leaves_alpha_diversity_treatment <- data.frame(leaves_alpha_diversity, Treatment = treatment_leaves_whole, Year = year_permanova_leaves)
leaves_alpha_diversity_treatment$X <- rownames(leaves_alpha_diversity_treatment)

leaves_alpha_diversity_treatment$X<-sub("18","",leaves_alpha_diversity_treatment$X)

leaves_alpha_diversity_treatment$X<-sub("19","",leaves_alpha_diversity_treatment$X)
 
library(tidyr)
leaves_alpha_diversity_wide<-pivot_wider(leaves_alpha_diversity_treatment, names_from = Year, values_from=leaves_alpha_diversity)

leaves_alpha_diversity_wide$yeardiffs<-leaves_alpha_diversity_wide$`2019`- leaves_alpha_diversity_wide$`2018`
hist(leaves_alpha_diversity_wide$yeardiffs)

t.test.year.diffs_leaves <-t.test(yeardiffs ~ Treatment,leaves_alpha_diversity_wide)
t.test.year.diffs_leaves

#Richness
leaves_richness_treatment <- data.frame(leaves_rich, Treatment = leaves_alpha_diversity_treatment$Treatment, Year = year_permanova_leaves)

#Delete the treatment column
leaves_richness_treatment <- leaves_richness_treatment[, -2]
colnames(leaves_richness_treatment)[2] <- "Treatment"

leaves_richness_treatment$Sample <- rownames(leaves_richness_treatment)
leaves_richness_treatment$Sample <- sub("18", "", leaves_richness_treatment$Sample)
leaves_richness_treatment$Sample <- sub("19", "", leaves_richness_treatment$Sample)

library(tidyr)
leaves_rich_wide <- pivot_wider(leaves_richness_treatment, names_from = Year, values_from = Richness)
leaves_rich_wide$yeardiffs <- leaves_rich_wide$`2019` - leaves_rich_wide$`2018`
hist(leaves_rich_wide$yeardiffs)

t_test_leaves_rich <- t.test(yeardiffs ~ Treatment, leaves_rich_wide)
t_test_leaves_rich
```

#T-test by year

```{r}
library(tidyr)
leaves_alpha_t_year <- leaves_alpha_diversity_treatment[, -2]
leaves_alpha_t_wide <- pivot_wider(leaves_alpha_t_year, names_from = Year, values_from = leaves_alpha_diversity)

t_leaves_alpha_year <- t.test(leaves_alpha_t_wide$`2019`, leaves_alpha_t_wide$`2018`, paired = TRUE, alternative = "two.sided")
t_leaves_alpha_year

#Richness
leaves_richness_t <- leaves_richness_treatment
leaves_richness_t <- leaves_richness_t[, -2]
leaves_richness_wide <- pivot_wider(leaves_richness_t, names_from = Year, values_from = Richness)

leaves_richness_t_year <- t.test(leaves_richness_wide$`2019`, leaves_richness_wide$`2018`, paired = TRUE, alternative = "two.sided")
leaves_richness_t_year
```

