---
title: "16s_soils"
author: "Beant Kapoor"
date: "3/6/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

### General Comment - Objects in this file contain the name 'otu' however they represent ASV in this case.

```{r global_options, include=FALSE}
knitr::opts_chunk$set(echo=FALSE, warning=FALSE, message=FALSE)
```

# The purpose of this code chunk is to find out the number of total sequences and ASVs without the fruit samples. This is going to be in the Bacterial Sequence Processing section in the Results.

```{r}
library(ggplot2)
library(vegan)
# We will be kicking out the samples belonging to the fruits niche
fruit_names <- c("18CF747", "18CF1517", "18CF1527","18CF1537","18CF1547","18CF1567", "18CF2217", "18CF2227", "18CF2237", "18CF2247", "18CF2257", "18CF3917", "18CF3927", "18CF3947", "18CF3967", "18CF3987", "PCRblank1", "PCRblank2")

whole_otu_without_fruits <- subset(whole_otu, !(rownames(whole_otu) %in% fruit_names))

#Let's remove the columns which are not needed
whole_otu_without_fruits <- whole_otu_without_fruits[, 3:170189]

#Let's remove the OTUs that is not represented in any of the samples.
whole_otu_without_fruits <- whole_otu_without_fruits[, colSums(whole_otu_without_fruits) > 0]

ncol(whole_otu_without_fruits) #Number of OTUs across 200 samples is 169,999. 
sum(whole_otu_without_fruits) #Number of sequences across 200 samples is 5,388,026.
```


```{r Import Data and Subset}
#First I import the whole OTU table I recieved from Mothur. Note at this point taxonomy is not included, it is just raw OTU counts (columns) by sample or group (rows). Below I indicate that the row.names are in column 2 of the imported file.
whole_otu <- read.table("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/files_from_mothur/16s_all.shared", header = TRUE, sep = '\t', fill = TRUE, strip.white = TRUE, row.names = 2)

#Next I import the metadata for the particular habitat of interest. In this case I am working with soils. I will use this metadata to subset the larger OTU file above.
soils_metadata <- read.table("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/metadata/16s_soils.txt", header = TRUE, fill = TRUE, strip.white = TRUE)
soil_sample_names <- soils_metadata$Sample

#Next I import the taxonomy file. This contains the taxonomy assignments for all OTUs in the study and not just the ones for the particuluar habitat of interest. This will get subset as well.
taxonomy <- read.table("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/files_from_mothur/16s_all.taxonomy", header = TRUE, sep = '\t', fill = TRUE, strip.white = TRUE, row.names = 1)

#Here I use the subset function to pull out just the samples belonging to soils.
soils_otu <- subset(whole_otu, rownames(whole_otu) %in% soil_sample_names)
```

```{r Data Clean-up}
#Let's remove the columns which are not needed
soils_otu_only <- soils_otu[, 3:170189]

#Here I am removing any OTU that is not present in any samples. What the below function is doing is summing the columns (OTUs) and any OTU that equals 0 is removed from the dataset. This will hopefully speed up downstream analyses.
soils_pure <- soils_otu_only[, colSums(soils_otu_only) > 0]
```

```{r Rarefaction}
#Below I create the rarefaction curves for each sample using the rarecurve function from vegan.
library(vegan)
rare_curve_soils <- rarecurve(soils_pure)

#I then create an object that spans the range of the ESV richness. This will be used to plot the vertical line in my ggplot rarefaction curve. 
f_soils<-(0:12500)

#I then create objects containing the x and y coordinates values for each stratum.
sequences_soils40<-attr(rare_curve_soils[[40]],which="Subsample")
sequences_soils39<-attr(rare_curve_soils[[39]],which="Subsample")
sequences_soils38<-attr(rare_curve_soils[[38]],which="Subsample")
sequences_soils37<-attr(rare_curve_soils[[37]],which="Subsample")
sequences_soils36<-attr(rare_curve_soils[[36]],which="Subsample")
sequences_soils35<-attr(rare_curve_soils[[35]],which="Subsample")
sequences_soils34<-attr(rare_curve_soils[[34]],which="Subsample")
sequences_soils33<-attr(rare_curve_soils[[33]],which="Subsample")
sequences_soils32<-attr(rare_curve_soils[[32]],which="Subsample")
sequences_soils31<-attr(rare_curve_soils[[31]],which="Subsample")
sequences_soils30<-attr(rare_curve_soils[[30]],which="Subsample")
sequences_soils29<-attr(rare_curve_soils[[29]],which="Subsample")
sequences_soils28<-attr(rare_curve_soils[[28]],which="Subsample")
sequences_soils27<-attr(rare_curve_soils[[27]],which="Subsample")
sequences_soils26<-attr(rare_curve_soils[[26]],which="Subsample")
sequences_soils25<-attr(rare_curve_soils[[25]],which="Subsample")
sequences_soils24<-attr(rare_curve_soils[[24]],which="Subsample")
sequences_soils23<-attr(rare_curve_soils[[23]],which="Subsample")
sequences_soils22<-attr(rare_curve_soils[[22]],which="Subsample")
sequences_soils21<-attr(rare_curve_soils[[21]],which="Subsample")
sequences_soils20<-attr(rare_curve_soils[[20]],which="Subsample")
sequences_soils19<-attr(rare_curve_soils[[19]],which="Subsample")
sequences_soils18<-attr(rare_curve_soils[[18]],which="Subsample")
sequences_soils17<-attr(rare_curve_soils[[17]],which="Subsample")
sequences_soils16<-attr(rare_curve_soils[[16]],which="Subsample")
sequences_soils15<-attr(rare_curve_soils[[15]],which="Subsample")
sequences_soils14<-attr(rare_curve_soils[[14]],which="Subsample")
sequences_soils13<-attr(rare_curve_soils[[13]],which="Subsample")
sequences_soils12<-attr(rare_curve_soils[[12]],which="Subsample")
sequences_soils11<-attr(rare_curve_soils[[11]],which="Subsample")
sequences_soils10<-attr(rare_curve_soils[[10]],which="Subsample")
sequences_soils9<-attr(rare_curve_soils[[9]],which="Subsample")
sequences_soils8<-attr(rare_curve_soils[[8]],which="Subsample")
sequences_soils7<-attr(rare_curve_soils[[7]],which="Subsample")
sequences_soils6<-attr(rare_curve_soils[[6]],which="Subsample")
sequences_soils5<-attr(rare_curve_soils[[5]],which="Subsample")
sequences_soils4<-attr(rare_curve_soils[[4]],which="Subsample")
sequences_soils3<-attr(rare_curve_soils[[3]],which="Subsample")
sequences_soils2<-attr(rare_curve_soils[[2]],which="Subsample")
sequences_soils1<-attr(rare_curve_soils[[1]],which="Subsample")

#Here I plot the rarefaction curves. 

library(ggplot2)
library(ggpubr)

rarefaction_chart_soils <- ggplot() +
  geom_line(aes(x=sequences_soils40,y=rare_curve_soils[[40]]),size=1)+
  geom_line(aes(x=sequences_soils39,y=rare_curve_soils[[39]]), size=1)+
  geom_line(aes(x=sequences_soils38,y=rare_curve_soils[[38]]),size=1)+
  geom_line(aes(x=sequences_soils37,y=rare_curve_soils[[37]]),size=1)+
  geom_line(aes(x=sequences_soils36,y=rare_curve_soils[[36]]),size=1)+
  geom_line(aes(x=sequences_soils35,y=rare_curve_soils[[35]]),size=1)+
  geom_line(aes(x=sequences_soils34,y=rare_curve_soils[[34]]),size=1)+
  geom_line(aes(x=sequences_soils33,y=rare_curve_soils[[33]]), size=1)+
  geom_line(aes(x=sequences_soils32,y=rare_curve_soils[[32]]),size=1)+
  geom_line(aes(x=sequences_soils31,y=rare_curve_soils[[31]]),size=1 )+
  geom_line(aes(x=sequences_soils30,y=rare_curve_soils[[30]]),size=1)+
  geom_line(aes(x=sequences_soils29,y=rare_curve_soils[[29]]),size=1)+
  geom_line(aes(x=sequences_soils28,y=rare_curve_soils[[28]]),size=1)+
  geom_line(aes(x=sequences_soils27,y=rare_curve_soils[[27]]),size=1)+
  geom_line(aes(x=sequences_soils26,y=rare_curve_soils[[26]]),size=1)+
  geom_line(aes(x=sequences_soils25,y=rare_curve_soils[[25]]), size=1)+
  geom_line(aes(x=sequences_soils24,y=rare_curve_soils[[24]]),size=1)+
  geom_line(aes(x=sequences_soils23,y=rare_curve_soils[[23]]),size=1)+
  geom_line(aes(x=sequences_soils22,y=rare_curve_soils[[22]]),size=1)+
  geom_line(aes(x=sequences_soils21,y=rare_curve_soils[[21]]),size=1)+
  geom_line(aes(x=sequences_soils20,y=rare_curve_soils[[20]]),size=1)+
  geom_line(aes(x=sequences_soils19,y=rare_curve_soils[[19]]), size=1)+
  geom_line(aes(x=sequences_soils18,y=rare_curve_soils[[18]]),size=1)+
  geom_line(aes(x=sequences_soils17,y=rare_curve_soils[[17]]),size=1)+
  geom_line(aes(x=sequences_soils16,y=rare_curve_soils[[16]]),size=1)+
  geom_line(aes(x=sequences_soils15,y=rare_curve_soils[[15]]),size=1)+
  geom_line(aes(x=sequences_soils14,y=rare_curve_soils[[14]]), size=1)+
  geom_line(aes(x=sequences_soils13,y=rare_curve_soils[[13]]), size=1)+
  geom_line(aes(x=sequences_soils12,y=rare_curve_soils[[12]]),size=1)+
  geom_line(aes(x=sequences_soils11,y=rare_curve_soils[[11]]),size=1 )+
  geom_line(aes(x=sequences_soils10,y=rare_curve_soils[[10]]),size=1)+
  geom_line(aes(x=sequences_soils9,y=rare_curve_soils[[9]]),size=1)+
  geom_line(aes(x=sequences_soils8,y=rare_curve_soils[[8]]),size=1)+
  geom_line(aes(x=sequences_soils7,y=rare_curve_soils[[7]]),size=1)+
  geom_line(aes(x=sequences_soils6,y=rare_curve_soils[[6]]),size=1)+
  geom_line(aes(x=sequences_soils5,y=rare_curve_soils[[5]]), size=1)+
  geom_line(aes(x=sequences_soils4,y=rare_curve_soils[[4]]),size=1)+
  geom_line(aes(x=sequences_soils3,y=rare_curve_soils[[3]]),size=1)+
  geom_line(aes(x=sequences_soils2,y=rare_curve_soils[[2]]),size=1)+
  geom_line(aes(x=sequences_soils1,y=rare_curve_soils[[1]]),size=1)+
  geom_vline(xintercept = 11677, linetype = "dashed")+
  ylim(0, 12000)+
  xlim(0, 150000)+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
       panel.border=element_rect(color="black", size=1, fill=NA))+
 theme(axis.title.x=element_text(size=14, face="bold"))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+ 
labs(x=("Sequencing Depth"), y=expression(bold(paste("OTU Richness", sep=""))))+
  labs(title = 'Soils 16S')
rarefaction_chart_soils

ggsave(filename = 'soils_16s_rarefaction_curve.png', path = "/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Results", rarefaction_chart_soils, dpi = 300, width = 5, height = 5, units = 'in')

#Here we are rarefying the data using the rrarefy function from vegan. I first use the sort function to determine the lowest number of sequences in a sample. 
sort(rowSums(soils_pure)) #11677

#Then using the rrarefy function from vegan, I rarefy my samples. The rarefaction cutoff or "sample" value is determined by evaluating my rarefaction curves and choosing the number of sequences that minimizes sample loss but maximizes sampling depth.
rare_otu_soils <- as.data.frame(rrarefy(soils_pure, sample = 11677)) #In this case we are not losing any samples therefore we do not need to perform any subsetting.

#I am now removing OTUs present in the OTU table that were only present in samples removed following rarefaction. These OTUs will have column sums of 0.
rare_otu_soils <- rare_otu_soils[, colSums(rare_otu_soils) > 0]

#The following line is for saving the rarefied OTU table. This file is later reloaded to be used in downstream analyses. This is done to maintain consistency of results because rarefying generates slightly different results each time.
#write.table(rare_otu_soils, "/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Tables_from_R/16s_soils_rarefied_otu.shared", sep = '\t', row.names = TRUE, col.names = TRUE)

#I am now removing singleton OTUs (OTUs with only 1 representative sequence following rarefaction). This is to limit the effects of rare species on our distance matrices used during ordination.  
rare_otu_soils_no_singletons <- rare_otu_soils[, colSums(rare_otu_soils) > 1]

#The following line is for saving the rarefied OTU table with singletons removed. This file is later reloaded to be used in downstream analyses. This is done to maintain consistency of results because rarefying generates slightly different results each time.
#write.table(rare_otu_soils_no_singletons, "/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Tables_from_R/16s_soils_rarefied_no_singletons.shared", sep = '\t', row.names = TRUE, col.names = TRUE)
```

```{r Relative abundance chart - Phylum}
#Now that samples have been rarefied I can perform alpha and beta diversity analyses on my data sets. To begin I first reformat my OTU table so that I can do some filtering and add taxonomy information.
rare_otu_soils_no_singletons <- read.table("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Tables_from_R/16s_soils_rarefied_no_singletons.shared", header = TRUE, sep = '\t')

rare_otu_soils <- read.table("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Tables_from_R/16s_soils_rarefied_otu.shared", header = TRUE, sep = '\t')

#I now transpose my OTU table so that I can begin to add in taxonomy information.
trans_otu_soils <- as.data.frame(t(rare_otu_soils_no_singletons))

#The below function is asking specifically for the OTU IDs in the soils OTU table. These are the row names. I will use these to subset the taxonomy file.
soils_otu_names <- rownames(trans_otu_soils)
soils_taxonomy <- subset(taxonomy, rownames(taxonomy) %in% soils_otu_names)

#Now I am subsetting the new taxonomy file to include only the taxonomy information stored in column 2. 
soils_taxonomy_only <- soils_taxonomy[, 2]

#Below I am using the separate function from tidyr to split the taxonomy strings into columns by semi-colon so that I can rename the OTUs to give them more meaning for downstream analyses.
library(tidyr)

#The SILVA database only provides classifications up to genus level, new columns are created only up to Genus level.
taxonomy_labels<-c("kingdom","phylum","class","order","family","genus")

#Below I create a new data frame with a column containing taxonomy information.
soils_otu_taxonomy <- data.frame(taxonomy = soils_taxonomy_only, trans_otu_soils)

#I then separate the taxonomy strings into new columns labeled with the taxonomy labels saved in the taxonomy_labels object. Strings are being separated based on the presence of semi-colons.
soils_otu_taxonomy <- separate(soils_otu_taxonomy, col = taxonomy, into = taxonomy_labels, sep = ';')

#Below I create a data frame that consists of phylum assignment and the rarefied OTU table.
soils_otu_phylum <- data.frame(phylum = soils_otu_taxonomy$phylum, trans_otu_soils)

#I then remove the assignment confidence values contained in parentheses found in each taxonomic string using the sub function.
soils_otu_phylum$phylum <- sub("\\(.*)","",x = soils_otu_phylum$phylum)

#The following code is for the generation of phylum relative abundance stacked bar charts. I first sum each OTU by the phylum it belongs to. 
library(plyr)
soils_summed_otu_phylum <- as.data.frame(ddply(soils_otu_phylum, .(phylum), colwise(sum)))
rownames(soils_summed_otu_phylum) <- make.names(soils_summed_otu_phylum$phylum)
soils_summed_otu_phylum <- soils_summed_otu_phylum[, 2:41]

#I then transpose the data frame so that sample names are now row names and column names are OTU names.
trans_soils_otu_summed_phylum <- as.data.frame(t(soils_summed_otu_phylum))
rownames(trans_soils_otu_summed_phylum) <- sub('X', '', x = rownames(trans_soils_otu_summed_phylum))

#Read in a new metadata file for soils to make relative abundance charts
install.packages("readxl")
library(readxl)
soils_metadata_for_rel_abund <- read_excel("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/metadata/16s_metadata_without_plot.xlsx")
trans_soils_phylum_metadata <- merge(soils_metadata_for_rel_abund, trans_soils_otu_summed_phylum, by.x = 1, by.y = 0)

#Then we can delete the sample name column as we don't need that.
trans_soils_phylum_metadata <- trans_soils_phylum_metadata[, 2:35]

#I then sum each OTU by Treatment column and subset to exclude sample names. This is because the following functions only work on matrices containing numeric data.
soils_phylum_summed_treatment <- as.data.frame(ddply(trans_soils_phylum_metadata, .(Treatment), colwise(sum)))
soils_phylum <- soils_phylum_summed_treatment[, 2:34]

#I am now creating an other category. Other includes those OTUs belonging to a phylum that comprises less than 1% of the total community. This new object is reffered to as phylum_others.
soils_phylum_others <- soils_phylum[, colSums(soils_phylum)/sum(soils_phylum) <= 0.01]
soils_phylum_others_sum <- rowSums(soils_phylum_others)

#I then create an object that contains all of the phyla that comprise more than 1% of the total community.
soils_phylum_regular <- soils_phylum[, colSums(soils_phylum)/sum(soils_phylum) > 0.01]

#I then create a dataframe containing the treatment information, the Other column and the remaining phyla
soils_phylum_reg_others <- data.frame(Treatment = soils_phylum_summed_treatment$Treatment, soils_phylum_regular, Others = soils_phylum_others_sum)

#These values are then converted to relative abundance using the decostand function from vegan. 
library(vegan)
soils_phylum_rel_abund <- decostand(soils_phylum_reg_others[, 2:10], method = 'total')
soils_phylum_rel_abund <- data.frame(Treatment = soils_phylum_reg_others$Treatment, soils_phylum_rel_abund)

#Below I use the melt function from the data.table package to reformat the data into stacked bar graph format. 
library(data.table)
library(ggplot2)
library(ggpubr)
library(RColorBrewer)
soils_phylum_melt <- melt(soils_phylum_rel_abund, id.vars = 'Treatment', variable.name = 'Phylum')
soils_colors_n_phylum <- length(unique(soils_phylum_melt[, 'Phylum']))
soils_phylum_melt$Treatment <- as.character(soils_phylum_melt$Treatment)
soils_phylum_melt$Treatment <- factor(soils_phylum_melt$Treatment, levels = c("Control 2018", "Control 2019", "Pre-burned 2018", "Post-burned 2019"))

tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Results/16s_soils_relative_abundance_phylum.tiff", units = 'in', width = 7, height = 7, res = 200)
soils_rel_chart_phylum <- ggplot(soils_phylum_melt, aes(x = Treatment, y = value, fill = Phylum))+
  geom_bar(stat = 'identity', show.legend = TRUE, color = 'black')+
  scale_fill_manual(values = colorRampPalette(brewer.pal(7, 'Accent'))(soils_colors_n_phylum))+
  xlab('Soils')+
  ylab('Relative Abundance')+
  ggtitle('Soil - Phylum level')+
  theme(panel.border = element_blank(),
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(), 
        axis.text.y = element_text(size = 14), 
        axis.line.y.left = element_line(), 
        axis.title.y = element_blank(), 
        legend.title = element_text(face = "bold", size = 14),
        legend.text = element_text(size = 14))
soils_rel_chart_phylum
```

#The purpose of the following chunk is to create a dataframe with realtive abundace in percentage for phylum

```{r}
soils_phylum_percent <- soils_phylum_rel_abund * 100 
soils_phylum_percent <- soils_phylum_percent[, -1]
soils_phylum_percent <- data.frame(Treatment = soils_phylum_rel_abund$Treatment, soils_phylum_percent)
is.num <- sapply(soils_phylum_percent, is.numeric)
soils_phylum_percent[is.num] <- lapply(soils_phylum_percent[is.num], round, 2)
soils_proteobac_pre_mean <- mean(soils_phylum_percent$Proteobacteria[5:8])
soils_proteobac_pre_mean
soils_proteobac_pre_std <- sd(soils_phylum_percent$Proteobacteria[5:8])
soils_proteobac_pre_std
soils_acidobac_post <- mean(soils_phylum_percent$Acidobacteriota[1:4])
soils_acidobac_post
```


# The following chunk creates a relative abundance chart at the class level.

```{r Relative abundance chart - Class}
library(plyr)
soils_otu_class <- data.frame(class = soils_otu_taxonomy$class, trans_otu_soils)

#I then remove the assignment confidence values contained in parentheses found in each taxonomic string using the sub function.
soils_otu_class$class <- sub("\\(.*)", '', x = soils_otu_class$class)
soils_summed_class <- as.data.frame(ddply(soils_otu_class, .(class), colwise(sum)))
rownames(soils_summed_class) <- make.names(soils_summed_class$class)
soils_summed_class <- soils_summed_class[, 2:41]

#I then transpose the dataframe and merge the metadata information
trans_soils_class <- as.data.frame(t(soils_summed_class))
rownames(trans_soils_class) <- sub('X', '', x = rownames(trans_soils_class))
soils_class_metadata <- merge(soils_metadata_for_rel_abund, trans_soils_class, by.x = 1, by.y = 0)

#Then we can delete the column with the sample names as they are not needed and then sum by treatment
soils_class_metadata <- soils_class_metadata[, 2:94]
soils_class_summed_treamtent <- as.data.frame(ddply(soils_class_metadata, .(Treatment), colwise(sum)))

#I then subset the data so that only numerical data is present and creating a new class of objects, other, that includes OTUs from classes that represent 1% or less of the total community.
soils_class <- soils_class_summed_treamtent[, 2:93]
soils_class_others <- soils_class[, colSums(soils_class)/sum(soils_class) <= 0.01]
soils_class_others_sum <- rowSums(soils_class_others)
soils_class_regular <- soils_class[, colSums(soils_class)/sum(soils_class) > 0.01]

#I then create a new dataframe that contains all of the class data and determine relative abundace using decostand function from vegan
soils_class_reg_others <- data.frame(Treatment = soils_class_summed_treamtent$Treatment, soils_class_regular, Others = soils_class_others_sum)
soils_class_rel_abund <- decostand(soils_class_reg_others[, 2:16], method = 'total')
soils_class_rel_abund <- data.frame(Treatment = soils_class_reg_others$Treatment, soils_class_rel_abund)

#Below I use the melt function from the data.table package to reformat the data into stacked bar graph format.
library(data.table)
library(ggplot2)
library(ggpubr)
library(RColorBrewer)
soils_class_melt <- melt(soils_class_rel_abund, id.vars = 'Treatment', variable.name = 'Class')
soils_class_n_color <- length(unique(soils_class_melt$Class))
soils_class_melt$Treatment <- as.character(soils_class_melt$Treatment)
soils_class_melt$Treatment <- factor(soils_class_melt$Treatment, levels = c("Control 2018", "Control 2019", "Pre-burned 2018", "Post-burned 2019"))

tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Results/16s_soils_relative_abundance_class.tiff", units = 'in', width = 7, height = 7, res = 200)
soils_rel_chart_class <- ggplot(soils_class_melt, aes(x = Treatment, y = value, fill = Class))+
  geom_bar(stat = 'identity', show.legend = TRUE, color = 'black')+
  scale_fill_manual(values = colorRampPalette(brewer.pal(7, 'Accent'))(soils_class_n_color))+
  xlab('Soils')+
  ylab('Relative Abundance')+
  ggtitle('Soil - Class level')+
  theme(panel.border = element_blank(), 
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.title.x = element_blank(), 
        axis.text.x = element_blank(), 
        axis.ticks.x = element_blank(), 
        axis.text.y = element_text(size = 14), 
        axis.line.y.left = element_line(), 
        axis.title.y = element_blank())
soils_rel_chart_class
dev.off()
```

#The purpose of the following chunk is to create a dataframe with realtive abundace in percentage for class

```{r}
soils_class_percent <- soils_class_rel_abund * 100 
soils_class_percent <- soils_class_percent[, -1]
soils_class_percent <- data.frame(Treatment = soils_class_rel_abund$Treatment, soils_class_percent)
is.num <- sapply(soils_class_percent, is.numeric)
soils_class_percent[is.num] <- lapply(soils_class_percent[is.num], round, 2)
soils_alphaproteo_pre_mean <- mean(soils_class_percent$Alphaproteobacteria[5:8])
soils_alphaproteo_pre_mean
soils_acidobacteriae_post <- mean(soils_class_percent$Acidobacteriae[1:4])
soils_acidobacteriae_post
```

# The following chunk joins the relative abundance chart at phylum and class level together.

```{r}
tiff(filename = "/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Results/16s_soils_phylum_class_rel_abund.tiff", units = 'in', width = 10, height = 7, res = 200)
ggarrange(soils_rel_chart_phylum, soils_rel_chart_class,  nrow = 1, labels = c("I", "II"), align = 'h')
dev.off()
```


# The following chunk creates the relative abundance chart for soils 16s niche at order level

```{r Relative abundace}
soils_otu_order <- data.frame(order = soils_otu_taxonomy$order, trans_otu_soils)

#I then remove the assignment confidence values contained in parentheses found in each taxonomic string using the sub function.
soils_otu_order$order <- sub('\\(.*)', '', x = soils_otu_order$order)
soils_summed_order <- as.data.frame(ddply(soils_otu_order, .(order), colwise(sum)))
rownames(soils_summed_order) <- make.names(soils_summed_order$order)
soils_summed_order <- soils_summed_order[, 2:41]

#I then transpose the dataframe and merge the metadata information
trans_soils_order <- as.data.frame(t(soils_summed_order))
rownames(trans_soils_order) <- sub('X', '', x = rownames(trans_soils_order))
soils_order_metadata <- merge(soils_metadata_for_rel_abund, trans_soils_order, by.x = 1, by.y = 0)
soils_order_metadata <- soils_order_metadata[ ,2:230]

#sum by treatment
soils_order_summed_treatment <- as.data.frame(ddply(soils_order_metadata, .(Treatment), colwise(sum)))

#I then subset the data so that only numerical data is present and creating a new class of objects, other, that includes OTUs from classes that represent 3% or less of the total community.
soils_order <- soils_order_summed_treatment[ , 2:229]
soils_order_others <- soils_order[, colSums(soils_order)/sum(soils_order) <= 0.03]
soils_order_others_sum <- rowSums(soils_order_others)
soils_order_regular <- soils_order[, colSums(soils_order)/sum(soils_order) > 0.03]

#I then create a new dataframe that contains all of the class data and determine relative abundace using decostand function from vegan
soils_order_reg_others <- data.frame(Treatment = soils_order_summed_treatment$Treatment, soils_order_regular, Others = soils_order_others_sum)
soils_order_rel_abund <- decostand(soils_order_reg_others[, 2:10], method = 'total')
soils_order_rel_abund <- data.frame(Treatment = soils_order_reg_others$Treatment, soils_order_rel_abund)

#Below I use the melt function from the data.table package to reformat the data into stacked bar graph format.
library(data.table)
library(ggplot2)
library(ggpubr)
library(RColorBrewer)
soils_order_melt <- melt(soils_order_rel_abund, id.vars = 'Treatment', variable.name = 'order')
soils_order_n_color <- length(unique(soils_order_melt$order))
soils_order_melt$Treatment <- as.character(soils_order_melt$Treatment)
soils_order_melt$Treatment <- factor(soils_order_melt$Treatment, levels = c('Pre_control_A', 'Post_control_A', 'Pre_control_D', 'Post_control_D', 'Pre_burn_B', 'Post_burn_B', 'Pre_burn_C', 'Post_burn_C'))

tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Results/16s_soils_relative_abundance_order.tiff", units = 'in', width = 7, height = 7, res = 200)
ggplot(soils_order_melt, aes(x = Treatment, y = value, fill = order))+
  geom_bar(stat = 'identity', show.legend = TRUE, color = 'black')+
  scale_fill_manual(values = colorRampPalette(brewer.pal(7, 'Accent'))(soils_order_n_color))+
  xlab('Soils')+
  ylab('Relative Abundance')+
  theme(panel.border = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.title.x = element_text(size = 16), axis.text.x = element_text(size = 14, color = 'black', angle = 90), axis.ticks.x = element_blank(), axis.text.y = element_text(size = 14), axis.line.y.left = element_line(), axis.title.y = element_text(size = 16))
dev.off()
```

# The purpose of the following chunk is to create Shannon Alpha diversity boxplots across treatments

```{r Alpha Diversity}
# We will calculate richness and create a dataframe including richness with metadata
# Calculate the species richness using the richness function of vegan. We are going to work with the object named 'rare_otu_soils' because the structure of this dataset behaves as vegan community matrix data.
# Some pre-processing first as always
rownames(rare_otu_soils) <- sub('X', '', rownames(rare_otu_soils))
colnames(rare_otu_soils) <- gsub('\\.[0-9][0-9].$', '', colnames(rare_otu_soils))

#The following function from vegan will give us the shannon diversity index. That rarefied OTU table is used which has the singletons to sampling depth standard between samples.
library(vegan)
soils_alpha_diversity <- diversity(rare_otu_soils, index = 'shannon')
soils_alpha_diversity <- as.data.frame(soils_alpha_diversity)

# Let's add treatment and year columns to the soils_alpha_diversity dataframe
treatment_soils_whole
year_permanova_soils
soils_alpha_diversity_treatment <- data.frame(Treatment = treatment_soils_whole, Year = year_permanova_soils, soils_alpha_diversity)

#Write this dataframe to csv file
#write.csv(soils_alpha_diversity_treatment, file = "/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Tables_from_R/16s_soils_shannon.csv")

#Adding in the metadata for graph
soils_alpha_metadata <- merge(soils_metadata_for_rel_abund, soils_alpha_diversity, by.x = 1, by.y = 0)
rownames(soils_alpha_metadata) <- make.names(soils_alpha_metadata$Sample)
soils_alpha_metadata <- soils_alpha_metadata[, 2:3]
rownames(soils_alpha_metadata) <- sub('X', '', rownames(soils_alpha_metadata))
soils_alpha_metadata$Treatment <- as.character(soils_alpha_metadata$Treatment)
soils_alpha_metadata$Treatment <- factor(soils_alpha_metadata$Treatment, levels = c('Pre_control_A', 'Post_control_A', 'Pre_control_D', 'Post_control_D', 'Pre_burn_B', 'Post_burn_B', 'Pre_burn_C', 'Post_burn_C'))

#Let's make the graph now
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Results/16s_soils_alpha_shannon_index.tiff", units = 'in', width = 10, height = 7, res = 200)
ggplot(soils_alpha_metadata, aes(x = Treatment, y = soils_alpha_diversity, fill = Treatment))+
  geom_boxplot(outlier.shape = NA)+
  ggtitle('Soils 16S Alpha Diversity')+
  ylab('Shannon Diversity Index')+
  xlab('Treatment')+
  theme_linedraw()+
  theme(axis.title.x = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(plot.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.x = element_text(family = 'Times New Roman', size = 14, colour = 'black', face = 'bold', angle = 90))+
  theme(axis.title.y = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.y = element_text(family = 'Times New Roman', size = 14, face = 'bold'))+
  theme(legend.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(legend.text = element_text(size = 18, family = 'Times New Roman'))+
  scale_fill_brewer(palette = 'Set1')+
  guides(fill=guide_legend(title = 'Treatment'))
dev.off()

#Let's make the graph now without legend
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Results/16s_soils_alpha_shannon_index_without_legend.tiff", units = 'in', width = 8, height = 7, res = 200)
soils_alpha_chart_without_legend <- ggplot(soils_alpha_metadata, aes(x = Treatment, y = soils_alpha_diversity, fill = Treatment))+
  geom_boxplot(outlier.shape = NA)+
  ggtitle('Soils 16S Alpha Diversity')+
  ylab('Shannon Diversity Index')+
  xlab('Treatment')+
  theme_linedraw()+
  theme(axis.title.x = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(plot.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.x = element_text(family = 'Times New Roman', size = 14, colour = 'black', face = 'bold', angle = 90))+
  theme(axis.title.y = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.y = element_text(family = 'Times New Roman', size = 14, face = 'bold'))+
  theme(legend.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(legend.text = element_text(size = 18, family = 'Times New Roman'))+
  theme(legend.position = 'none')+
  scale_fill_brewer(palette = 'Set1')+
  guides(fill=guide_legend(title = 'Treatment'))
soils_alpha_chart_without_legend
dev.off()

```

# The following code is used to generate the Principal Coordinate Analysis chart

```{r Principal Coordinate Analyses}
#We will start with the object named 'rare_otu_soils_no_singletons'
library(vegan)
rare_soils_rel_abund_for_pcoa <- decostand(rare_otu_soils_no_singletons, method = 'total')

#Need to load the library 'ape' and use the vegdist function to create a dissimilarity matrix using the Bray Curtis distance
library(ape)
soils_dist <- vegdist(rare_soils_rel_abund_for_pcoa, method = 'bray')
soils_pcoa <- pcoa(soils_dist)
soils_pcoa_vectors <- as.data.frame(soils_pcoa$vectors)
soils_pcoa_vectors_metadata <- merge(soils_metadata_for_rel_abund, soils_pcoa_vectors, by.x = 1, by.y = 0)
biplot(soils_pcoa, plot.axes = c(1, 2))
soils_pcoa_scores <- data.frame(PC1 = soils_pcoa_vectors$Axis.1, PC2 = soils_pcoa_vectors$Axis.2)
soils_pcoa_ellipse <- ordiellipse(soils_pcoa_scores, soils_pcoa_vectors_metadata$Treatment)

#Plot the eigenvalues and interpret
barplot(soils_pcoa$values$Relative_eig[1:10])

# Find out the percentage variance explained by the first two principal axes
soils_pcoa$values$Relative_eig[1] * 100 #40.61
soils_pcoa$values$Relative_eig[2] * 100 #7.39

#Extract the plot scores from first two PCoA axes
soils_pcoa_axes <- soils_pcoa$vectors[, c(1, 2)]

#In order to change the color scheme of PCoA we will have to add one more column to the dataframe 'soils_pcoa_vectors_metadata'
soils_treatment_pcoa_color <- c('Burn_B', 'Burn_B','Burn_B','Burn_B','Burn_B', 'Burn_C', 'Burn_C','Burn_C','Burn_C','Burn_C', 'Control_D', 'Control_D','Control_D','Control_D','Control_D', 'Control_A', 'Control_A','Control_A','Control_A','Control_A')
soils_pcoa_vectors_metadata <- data.frame(Treatment_color = soils_treatment_pcoa_color, soils_pcoa_vectors_metadata)

tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Results/16s_soils_PCOA.tiff", units = 'in', width = 10, height = 7, res = 200)
soils_pcoa_16s <- ggplot(soils_pcoa_scores, aes(PC1, PC2, color = soils_pcoa_vectors_metadata$Treatment_color))+
  geom_point(size = 2, aes(shape = year_permanova_soils))+
  labs(shape = 'Year')+
  ggtitle('Soil 16S')+
  xlab('Principal Axis 1 (40.61%)')+
  xlim(c(-0.6, 0.6))+
  ylab('Principal Axis 2 (7.39%)')+
  theme_linedraw()+
  #scale_color_discrete(name = 'Niches')+
  theme(plot.title = element_text(family = 'Arial', size=14, face="bold"))+
  theme(axis.title.x=element_text(family = 'Arial', size=10))+
  theme(axis.title.y=element_text(family = 'Arial', size=10))+
  theme(axis.text.x=element_text(family = 'Arial', size=10))+
  theme(axis.text.y=element_text(family = 'Arial', size=10))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 14))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))+
  scale_color_brewer(palette = 'Set1', name = 'Treatment')
soils_pcoa_16s
dev.off()
```

# The purpose of the following code is to visualize the distibution of the genus Arthrobacter in the soil microbiome. Arthrobacter is a plant growth promoting bacteria and is often found in burned forests.

```{r}
# We will start working with the objects 'soils_otu_taxonomy' and 'trans_otu_soils'
soils_genus <- data.frame(Genus = soils_otu_taxonomy$genus, trans_otu_soils)

# We then remove the assignment confidence values contained in parentheses found in the phylum column using the sub function.
soils_genus$Genus <- sub("\\(.*)", '', x = soils_genus$Genus)

# Then we have to sum each OTU by the genus it belongs to
soils_summed_genus <- ddply(soils_genus, .(Genus), colwise(sum))

# Extact the row which belongs to the genus arthrobacter
soils_arthrobacter <- soils_summed_genus[grep('Arthrobacter', soils_summed_genus$Genus), ]

# Transpose the dataframe to generate a histogram
t_soils_arthrobacter <- as.data.frame(t(soils_arthrobacter))

# Remove the first row and change the column name to arthrobacter
t_soils_arthrobacter <- as.data.frame(t_soils_arthrobacter[2:41, 1])
colnames(t_soils_arthrobacter) <- 'arthrobacter'

#Generate the histogram
hist(as.numeric(t_soils_arthrobacter$arthrobacter), breaks = 30)

####CONCLUSION - No Arthrobacter OTUs were found in soils bacterial dataset
```

# The purpose of the following code is to perform PERMANOVA between control and burned plots for 2019. That means our POST-FIRE DATA ONLY

```{r}
# We have to subset the dataframe rare_soils_rel_abund_for_pcoa to extract the data from 2019 only
soil_rel_abund_2019 <- rare_soils_rel_abund_for_pcoa[21:40, ]
soil_treatment_2019 <- c('Burned','Burned','Burned','Burned','Burned','Burned','Burned','Burned','Burned','Burned', 'Control','Control','Control','Control','Control','Control','Control','Control','Control','Control')
soil_rel_abund_2019 <- data.frame(Treatment = soil_treatment_2019, soil_rel_abund_2019)

# Now we will perform PERMANOVA by treatment using the adonis function from vegan
library(vegan)
soils_permanova_2019 <- adonis(soil_rel_abund_2019[, -1] ~ soil_rel_abund_2019$Treatment, method = 'bray', permutations = 10000)
soils_permanova_2019
```

# The purpose of this code is to do PERMANOVA on the WHOLE data

```{r}
# PERMANOVA without plot information
treatment_soils_whole <- c("Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned","Unburned")

year_permanova_soils <- c("2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019")

soils_permanova_whole <- adonis(rare_soils_rel_abund_for_pcoa~treatment_soils_whole + year_permanova_soils + treatment_soils_whole * year_permanova_soils, method = 'bray', permutations = 10000)
soils_permanova_whole

soils_plot <- c("B","B","B","B","B","C","C","C","C","C","D","D","D","D","D","A","A","A","A","A","B","B","B","B","B","C","C","C","C","C","D","D","D","D","D","A","A","A","A","A")

#PERMANOVA with plot information
soils_permanova_whole_plot <- adonis(rare_soils_rel_abund_for_pcoa ~ treatment_soils_whole +  year_permanova_soils + soils_plot + treatment_soils_whole * year_permanova_soils * soils_plot, method = 'bray', permutations = 10000)
soils_permanova_whole_plot
```

#The purpose of this code chunk is to find differences in shannon alpha diversity across treatments.

```{r Differences in shannon index}
#We have to generate a new object for plot information
plot_tree_soils <- c("B_1", "B_2","B_3","B_4","B_5", "C_1", "C_2","C_3","C_4","C_5", "D_1", "D_2", "D_3", "D_4", "D_5", "A_1", "A_2","A_3","A_4","A_5","B_1", "B_2","B_3","B_4","B_5", "C_1", "C_2","C_3","C_4","C_5", "D_1", "D_2", "D_3", "D_4", "D_5", "A_1", "A_2","A_3","A_4","A_5")

#We will start with object 'soils_alpha_diversity_treatment' and add one more column to it and that is plot information
soils_alpha_diversity_treatment <- read.csv("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Tables_from_R/16s_soils_shannon.csv")
soils_plot_shannon <- data.frame(Plot = plot_tree_soils, soils_alpha_diversity_treatment)
colnames(soils_plot_shannon)[5] <- "Shannon_index"

#Install required packages if they are not already installed
if(!require(psych)){install.packages("psych")}
if(!require(nlme)){install.packages("nlme")}
if(!require(car)){install.packages("car")}
if(!require(multcompView)){install.packages("multcompView")}
if(!require(lsmeans)){install.packages("lsmeans")}
if(!require(ggplot2)){install.packages("ggplot2")}
if(!require(rcompanion)){install.packages("rcompanion")}

#Order factors by order in dataframe, otherwise, R will alphabetize them
soils_plot_shannon$Treatment <- factor(soils_plot_shannon$Treatment, levels = unique(soils_plot_shannon$Treatment))

#Convert the Year column to numeric
soils_plot_shannon$Year <- as.numeric(soils_plot_shannon$Year)

#This code is just to check if burn treatment has any effect on response varaible
library(nlme)
library(car)
soils_alpha_model_effect <- lme(Shannon_index ~ Treatment + Year + Treatment*Year, random = ~1|Plot, data = soils_plot_shannon, method = "REML")
Anova(soils_alpha_model_effect)

#Let's perform the tukey test on soils alpha diversity
library(lsmeans)
library(multcomp)
soils_alpha_post_hoc <- lsmeans(soils_alpha_model_effect, ~Year, adjust = "tukey")
cld(soils_alpha_post_hoc, alpha = 0.05, Letters = letters, adjust = "tukey")
```

#The purpose of this code is to find the differences in observed richness across treatments.

```{r Differences in Richness}
# RICHNESS
library(vegan)
soils_richness <- specnumber(rare_otu_soils)
soils_richness_metadata <- data.frame(Plot = plot_tree_soils, Year = year_permanova_soils, Treatment = treatment_soils_whole, Richness = soils_richness)

#Order factors by order in dataframe, otherwise, R will alphabetize them
soils_richness_metadata$Treatment <- factor(soils_richness_metadata$Treatment, levels = unique(soils_richness_metadata$Treatment))

#Make the Year column numeric
soils_richness_metadata$Year <- as.numeric(soils_richness_metadata$Year)

#This code is just to check if burn treatment has any effect on richness varaible
soils_richness_model_effect <- lme(Richness ~ Treatment + Year + Treatment*Year, random = ~1|Plot, correlation = corAR1(form = ~ Year | Plot, value = 0.4287), data = soils_richness_metadata, method = "REML")
Anova(soils_richness_model_effect)

#Code for the analysis
library(nlme)
soils_richness_metadata$treatment_year <- interaction(soils_richness_metadata$Treatment, soils_richness_metadata$Year)
soils_richness_model_tukey <- lme(Richness ~ treatment_year, random = ~1|Plot, correlation = corAR1(form = ~ Year | Plot, value = 0.4287), data = soils_richness_metadata, method = "REML")
Anova(soils_richness_model_tukey)

#Let's perform the tukey test on soils alpha diversity
library(lsmeans)
soils_richness_post_hoc <- lsmeans(soils_richness_model_tukey, ~treatment_year, adjust = "tukey")
cld(soils_richness_post_hoc, alpha = 0.05, Letters = letters, adjust = "tukey")

#One thing that we forgot is to make charts for the richness. We will do that now. Let's create the data frame required.
soils_richness_df <- data.frame(soils_alpha_metadata, Richness = soils_richness)
soils_richness_df <- soils_richness_df[, -2]
soils_richness_df$Treatment <- as.character(soils_richness_df$Treatment)
soils_richness_df$Treatment <- factor(soils_richness_df$Treatment, levels = c('Pre_control_A', 'Post_control_A', 'Pre_control_D', 'Post_control_D', 'Pre_burn_B', 'Post_burn_B', 'Pre_burn_C', 'Post_burn_C'))

# This is the graph with legend
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Results/16s_soils_richness.tiff", units = 'in', width = 10, height = 7, res = 200)
ggplot(soils_richness_df, aes(x = Treatment, y = Richness, fill = Treatment))+
  geom_boxplot(outlier.shape = NA)+
  ggtitle('Soils 16S Richness')+
  ylab('OTU Richness')+
  xlab('Treatment')+
  theme_linedraw()+
  theme(axis.title.x = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(plot.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.x = element_text(family = 'Times New Roman', size = 14, colour = 'black', face = 'bold', angle = 90))+
  theme(axis.title.y = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.y = element_text(family = 'Times New Roman', size = 14, face = 'bold'))+
  theme(legend.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(legend.text = element_text(size = 18, family = 'Times New Roman'))+
  scale_fill_brewer(palette = 'Set1')+
  guides(fill=guide_legend(title = 'Treatment'))
dev.off()

#Let's make the graph without legend
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Results/16s_soils_richness_without_legend.tiff", units = 'in', width = 8, height = 7, res = 200)
soils_richness_chart_without_legend <- ggplot(soils_richness_df, aes(x = Treatment, y = Richness, fill = Treatment))+
  geom_boxplot(outlier.shape = NA)+
  ggtitle('Soils 16S Richness')+
  ylab('OTU Richness')+
  xlab('Treatment')+
  theme_linedraw()+
  theme(axis.title.x = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(plot.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.x = element_text(family = 'Times New Roman', size = 14, colour = 'black', face = 'bold', angle = 90))+
  theme(axis.title.y = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(axis.text.y = element_text(family = 'Times New Roman', size = 14, face = 'bold'))+
  theme(legend.title = element_text(family = 'Times New Roman', face = 'bold', size = 18))+
  theme(legend.text = element_text(size = 18, family = 'Times New Roman'))+
  theme(legend.position = 'none')+
  scale_fill_brewer(palette = 'Set1')+
  guides(fill=guide_legend(title = 'Treatment'))
dev.off()

#Now let's combine the alpha diversity and richness chart using ggarrange
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Results/16s_soils_richness_and_shannon.tiff", units = 'in', height = 10, width = 8, res = 300)
ggarrange(soils_richness_chart_without_legend + theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()), soils_alpha_chart_without_legend,  ncol = 1, labels = c("I", "II"), align = 'v', hjust = -3)
dev.off()
```

```{r Concise box plots}
#The purpose of this code chunk is to make the shannon diversity box plots in the concise form. We will start working with dataframe soils_alpha_diversity
soils_16s_alpha_diversity <- soils_alpha_diversity
colnames(soils_16s_alpha_diversity)[1] <- 'values'

#Note that the metadata that we wanna use now has already been created for pcoa plot so will use that metadata.
soils_treatment_pcoa_color
year_permanova_soils
soils_16s_alpha_concise <- data.frame(soils_16s_alpha_diversity, Treatment = soils_treatment_pcoa_color, Year = year_permanova_soils)
soils_16s_alpha_concise$Treatment <- as.character(soils_16s_alpha_concise$Treatment)
soils_16s_alpha_concise$Treatment <- factor(soils_16s_alpha_concise$Treatment, levels = c('Control_A', 'Control_D', 'Burn_B', 'Burn_C'))

soils_16s_alpha_graph <- ggplot(data = soils_16s_alpha_concise, aes(x = Treatment, y = values)) + geom_boxplot(aes(fill = Year)) +
  ggtitle('Soils 16S')+
  xlab('Treatment')+
  ylab('Shannon Diversity Index')+
  theme_linedraw()+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  theme(axis.text.x=element_text(family = 'Arial', size=12))+
  theme(axis.text.y=element_text(family = 'Arial', size=12))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))+
  scale_color_brewer(palette = 'Set1', name = 'Treatment')

#We have to do the same for the richness graph. We will use soils_richness_df dataframe
soils_16s_richness <- soils_richness_df

#We wanna delete the existing Treatment column
soils_16s_richness <- subset(soils_16s_richness, select = -Treatment)

#Add the same metadata previously added
soils_16s_richness <- data.frame(soils_16s_richness, Treatment = soils_treatment_pcoa_color, Year = year_permanova_soils)
soils_16s_richness$Treatment <- as.character(soils_16s_richness$Treatment)
soils_16s_richness$Treatment <- factor(soils_16s_richness$Treatment, levels = c('Control_A', 'Control_D', 'Burn_B', 'Burn_C'))

soils_16s_richness_graph <- ggplot(data = soils_16s_richness, aes(x = Treatment, y = Richness)) + geom_boxplot(aes(fill = Year)) +
  #ggtitle('Soils 16S')+
  xlab('Treatment')+
  ylab('OTU Richness')+
  theme_linedraw()+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  scale_y_continuous(position = "right")+
  theme(axis.text.x=element_text(family = 'Arial', size=12))+
  theme(axis.text.y=element_text(family = 'Arial', size=12))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))+
  scale_color_brewer(palette = 'Set1', name = 'Treatment')
```

#The purpose of this code is to join the relative abundance charts for below ground niche together

```{r 16S soils and roots relative abundance}
#Let's join the relative abundance charts using ggarrange
library(ggpubr)
soils_roots_16s_rel <- ggarrange(soils_rel_chart_phylum, soils_rel_chart_class, roots_rel_chart_phylum, roots_rel_chart_class, nrow = 2, ncol = 2, labels = c("A", "B", "C", "D"), align = 'v', common.legend = FALSE)

#Annotate the figure
soils_roots_16s_rel <- annotate_figure(soils_roots_16s_rel,
            bottom = text_grob('Treatment', size = 14),
            left = text_grob('Relative abundance', size = 14, rot = 90))

#Save the image
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/bacteria_fungi/Results/16s_belowground_niches_relative_abundance_without_plot.tiff", units = 'in', width = 10, height = 10, res = 300)
soils_roots_16s_rel
dev.off()
```

```{r Soil Nutrient Analysis - PCA}
#Read in the metadata
soils_nutrients <- read.csv("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Soils_Nutrient_Analysis/soil_nutrient_metadata.csv", header = TRUE, row.names = 1, sep = ',', fill = TRUE, strip.white = TRUE)

#Delete the existing Treatment column and add properly formatted Treatment and Year column
soils_nutrients <- soils_nutrients[, c(2:16)]
treatment_soils_nutrients <- c("Control 2018", "Control 2018","Control 2018","Control 2018","Control 2018", "Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2019", "Control 2019","Control 2019","Control 2019","Control 2019", "Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019")
soils_nutrients <- data.frame(Treatment = treatment_soils_nutrients, Year = year_permanova_soils, soils_nutrients)
soils_nutrients$Treatment <- factor(soils_nutrients$Treatment, levels = c("Control 2018", "Control 2019", "Pre-burned 2018", "Post-burned 2019"))

#Let's conduct the PCA using the prcomp function. Prior to performing the PCA, all the variabes should be scaled and centered. First create the object containing only soils nutrients data.
soil_properties <- soils_nutrients[, c(3:17)]
summary(soil_properties)

#Scale and center variables
scaled_soil_properties <- scale(soil_properties, scale = TRUE, center = TRUE)
summary(scaled_soil_properties)

#Set seed to get consistent point placement in ordination space
set.seed(786)

#Perform PCA using prcomp function of factoextra package. Scale and Center are set to false since variables have already been scaled and centered.
install.packages('factoextra')
library(factoextra)
soils_pca <- prcomp(scaled_soil_properties, scale = FALSE, center = FALSE)

#Use summary to observe axis scores
summary(soils_pca)

#PCA evlaution. We can now evaluate the PCA using a series of plots generated by the factoextra package and corrplot packages.

#Scree plot
fviz_eig(soils_pca)

#PCA biplot
fviz_pca_biplot(soils_pca, label = 'none')

#Visualize contribution of each variable to specific dimensions (bar chart)
fviz_contrib(soils_pca, choice = 'var', axes = 1, top = 10)
fviz_contrib(soils_pca, choice = 'var', axes = 2, top = 10)
fviz_contrib(soils_pca, choice = 'var', axes = 3, top = 10)

fviz_pca_var(soils_pca, col.var = 'contrib', repel = TRUE)

#Load the corrplot package
install.packages('corrplot')
library(corrplot)

#Extract PCA variables
var <- get_pca_var(soils_pca)

#Visualize quality of variable representation
corrplot(var$cos2, is.corr = FALSE)

#Visualize contribution of each variable to each dimension
corrplot(var$contrib, is.corr = FALSE)

#Visualize correlation of each variable to each dimension
corrplot(var$cor, is.corr = FALSE)

#PCA Plot. The following code describes the generation of ellipses and plotting of PCA biplot using the ggbiplot function.

#PCA Ellipses
# First extract site scores using the axes you have identified as important
soils_pc1 <- soils_pca$x[1:40, 1]
soils_pc2 <- soils_pca$x[1:40, 2]
soil_pca_axes <- data.frame(cbind(soils_pc1, soils_pc2))

#Create a dataframe with axes scores and metadata
soils_pca_dataframe <- data.frame(Treatment = soils_nutrients$Treatment, Year = soils_nutrients$Year, PC1 = soil_pca_axes$soils_pc1, PC2 = soil_pca_axes$soils_pc2)

#Create ellipses using the ordiellipse function from vegan
library(vegan)
soils_pca_ellipse <- ordiellipse(soil_pca_axes, soils_pca_dataframe$Treatment, display = 'sites', kind = 'sd', draw = 'none')

df_ell <- data.frame()
for(g in levels(soils_pca_dataframe$Treatment)){
  df_ell <- rbind(df_ell, cbind(as.data.frame(with(soils_pca_dataframe[soils_pca_dataframe$Treatment == g, ], vegan:::veganCovEllipse(soils_pca_ellipse[[g]]$cov, soils_pca_ellipse[[g]]$center, soils_pca_ellipse[[g]]$scale))), Treatment = g))
}

soils_nutrient_color <- length(unique(df_ell$Treatment))
library(ggbiplot)
library(RColorBrewer)
soils_pca_nutrients <- ggbiplot(soils_pca, groups = soils_nutrients$Treatment, obs.scale =1, pc.biplot = TRUE)+
geom_path(data = df_ell, aes(x = soils_pc1, y = soils_pc2, colour = Treatment), size=1, linetype=2)+
  scale_fill_manual(values = colorRampPalette(brewer.pal(7, 'Accent'))(soils_nutrient_color))+
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(axis.title.x=element_text(size=14, face="bold"))+
  scale_x_continuous(breaks=c(-7,-3.5,0,3.5,7),limits=c(-7,7))+
  scale_y_continuous(breaks=c(-7,-3.5,0,3.5,7),limits=c(-7,7))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))+ 
  theme(panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(),
        panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(panel.grid.major=element_blank(),
       panel.grid.minor=element_blank(),
       panel.background = element_blank(),
       panel.border=element_rect(color="black", size=1, fill=NA))+
  labs(color="Treatment")+
  theme(legend.title = element_text(face = "bold", size = 14),
        legend.text = element_text(size = 14))

#Save the plot 
tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Results/soils_pca_nutrients.tiff", units = 'in', width = 7, height = 5, res = 300)
soils_pca_nutrients
dev.off()

#PERMANOVA - HAVE TO DO IT AGAIN!
simple_treatment <- c("Control","Control","Control","Control","Control","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Control","Control","Control","Control","Control","Control","Control","Control","Control","Control","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Control","Control","Control","Control","Control")
adonis(soils_nutrients[, 3:17] ~ simple_treatment + soils_nutrients$Year + simple_treatment * soils_nutrients$Year, method = 'euclidean', permutations = 10000)
```

```{r Anova test on 2018 data}
# We will start with subsetting the soils_alpha_diversity_treatment dataframe
soils_alpha_2018 <- as.data.frame(soils_alpha_diversity_treatment[1:20, 4])
colnames(soils_alpha_2018)[1] <- 'alpha_diversity'

plot_soils_anova <- c("B", "B","B","B","B","C","C","C","C","C","D","D","D","D","D","A","A","A","A","A")
soils_alpha_2018 <- data.frame(Plot = plot_soils_anova, soils_alpha_2018)

soils_alpha_2018_anova <- aov(alpha_diversity ~ Plot, data = soils_alpha_2018)
summary(soils_alpha_2018_anova)
TukeyHSD(soils_alpha_2018_anova)

#Let's do the PERMANOVA on the soils nutrients. We will subset the soils_nutrients dataframe.
soils_nutrients_2018 <- soils_nutrients[1:20, 3:17]

plot_permanova_soils <- c("A","A","A","A","A","B","B","B","B","B","C","C","C","C","C","D","D","D","D","D")
soils_nutrients_2018 <- data.frame(Plot = plot_permanova_soils, soils_nutrients_2018)

adonis(soils_nutrients_2018[, 2:15] ~ soils_nutrients_2018$Plot, method = 'euclidean', permutations = 10000)
```

#To identify the role of soil physicochemical variables in driving differences in soil bacterial community composition I perform a Constrained Correspondence Analysis (CCA). TO perform the CCA, I import the soil physicochemical data, assess soil matrix for multi-collinearity, perform model selection, construct a CCA biplot.

#Soil Data Import and OTU Table merging

##I will start with soils_nutrients and rare_otu_soils dataframe

```{r Constrained Correspondence Analysis (CCA)}
soils_nutrients
rare_otu_soils

#In the soils_nutrients dataframe, let's replace the underscores in rownames with nothing
rownames(soils_nutrients) <- gsub('_', '', rownames(soils_nutrients))

#Just realized that we have to add "1" at the end of the rownames of the soils_nutrients dataframe
rownames(soils_nutrients) <- paste(rownames(soils_nutrients), "1", sep = "")

#let's merge these two dataframes using the rownames
soils_otu_nutrients <- merge(soils_nutrients, rare_otu_soils, by = 0)

#CCA and Model Selection. I then perform the CCA using cca function from vegan. Prior to performing the CCA, I remove the variables that are strongly positively correlated. I then perform the model selection using the ordistep function to determine which variables to retain in the model.

#Below I create a matrix that consists of only soil physicochemical properties
soils_physicochemical <- data.frame(soils_nutrients[, 3:15])

#I then calculate the correlations for each soil variable to evaluate multicollinearity within the data
soils_corrtest <- cor(soils_physicochemical, method = "pearson")

#I plot the correlations using the corrplot function
library(corrplot)
corrplot(soils_corrtest, method = "number")

#Looking at the correlation matrix we can keep pH, S, Zn, Mn and NH.N. All other variables are highly correlated with these major drivers of the bacterial community composition.
soils_important_physicochemical <- data.frame(soils_nutrients[ , c(2, 4, 11, 13, 15)])

#Let's do the correlation test on the above matrix too
soils_imp_corrtest <- cor(soils_important_physicochemical, method = "pearson")
corrplot(soils_imp_corrtest, method = "number") #Looks good.

#I then create a matrix that consists only of the OTU table (must be numeric only)
soils_prop_otu <- soils_otu_nutrients[, 17:27224]

library(vegan)
#Set seed to get consistent point placement in ordination space
set.seed(786)

#Perform CCA with cca function from vegan. The first CCA is the intercept only and the second is the full model. These are needed for the model selection in subsequent steps.
soils_cca_intercept <- cca(soils_prop_otu ~ 1, data = soils_important_physicochemical)
soils_ccaplot <- cca(soils_prop_otu ~ ., data = soils_important_physicochemical)

#I evaluate multicollinearity one more time using the vif.cca function
vif.cca(soils_ccaplot)

#I then perform forward model selection using the ordistep function. An alternative method is to both directions or use the ordiR2step function.
soils_cca_model <- ordistep(soils_cca_intercept, scope = soils_ccaplot, permutations = how(nperm = 999), trace = TRUE, direction = "forward")

#To determine if soil physicochemical properties significantly explains the bacterial community perform permutation test with anova function. By including the 'by' argument we can identify significance of axes and model terms.
anova(soils_cca_model, steps = 10000)

#By axis
anova(soils_cca_model, steps = 10000, by = "axis")

#By terms
anova(soils_cca_model, steps = 10000, by = "terms")

# To obtain the information on axes, vectors etc. obtain CCA summary
soils_cca_model_summary <- summary(soils_cca_model)

#CCA Plot. Then to plot the CCA, I use the ggplot function. To use ggplot I need to first pull out the site scores for each sample, arrow coordinates, and generate ellipses.

#Since vegna does not feed directly into ggplot you will need to extract scores for sites and vectors to make CCAs in gggplot. Begin by extracting site scores (these will be the x and y coordinates for each sample)
soils_cca_site_scores <- as.data.frame(scores(soils_cca_model_summary, display = 'sites'))

#Generate some metadata to add to the dataframe
year_cca_soils <- c("2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2018","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019","2019")

#Create a new dataframe that includes the site scores from above with metadata from your study
soils_cca_graph <- data.frame(soils_cca_site_scores, CCA1 = soils_cca_site_scores$CCA1, CCA2 = soils_cca_site_scores$CCA2, Treatment = treatment_soils_nutrients, Year = year_cca_soils)

#To create a biplot you need to extract vector scores as below. To extract vector scores you need to use the object that contains the summary for CCA
soils_cca_arrows <- as.data.frame(soils_cca_model_summary$biplot)

#Then you need to create an object that contains the labels for the soil arrows and the coordinates for the soil arrows
soils_cca_labels <- labels(soils_cca_arrows)

#Here I am pulling out the arrow scores that are significant
soils_cca_arrows_ph1 <- soils_cca_arrows$CCA1[[c(1)]]
soils_cca_arrows_ph2 <- soils_cca_arrows$CCA2[[c(1)]]
soils_cca_arrows_ph <- data.frame(CCA1 = soils_cca_arrows_ph1, CCA2 = soils_cca_arrows_ph2)

soils_cca_arrows_mn1 <- soils_cca_arrows$CCA1[[c(2)]]
soils_cca_arrows_mn2 <- soils_cca_arrows$CCA2[[c(2)]]
soils_cca_arrows_mn <- data.frame(CCA1 = soils_cca_arrows_mn1, CCA2 = soils_cca_arrows_mn2)

#Here you determine the scaling factor for geom segment and geom point
soils_cca_arrowmult <- ordiArrowMul(soils_cca_model_summary, display = "sites")

#This is where you make confidence ellipses. I don't know what all the code means. Just know where to plug in my objects
soils_cca_ellipse <- ordiellipse(soils_cca_site_scores, soils_cca_graph$Treatment, display = "sites", kind = "sd", draw = "none")

df_ell_cca <- data.frame()
for(g in levels(soils_cca_graph$Treatment)){
  df_ell_cca <- rbind(df_ell_cca, cbind(as.data.frame(with(soils_cca_graph[soils_ccaplot$Treatment == g, ], vegan:::veganCovEllipse(soils_cca_ellipse[[g]]$cov, soils_cca_ellipse[[g]]$center, soils_cca_ellipse[[g]]$scale))), Treatment = g))
}

soils_cca_color <- length(unique(df_ell_cca$Treatment))
#Plot CCA in ggplot
library(RColorBrewer)
library(ggplot2)

soils_cca_chart <- ggplot(soils_cca_graph, aes(CCA1, CCA2, colour = Treatment))+
  geom_point(size = 3)+
  geom_path(data = df_ell_cca, aes(x = CCA1, y = CCA2, colour = Treatment), size = 1, linetype = 2)+
  geom_text(data = soils_cca_arrows_ph, aes(x = 2*CCA1, y = 2*CCA2, label = 'pH', fontface = "bold"), inherit.aes = FALSE)+
  geom_segment(data = soils_cca_arrows_ph, aes(x = 0, xend = 1.742239*CCA1, y = 0, yend = 1.742239 * CCA2), arrow = arrow(length = unit(0.25, "cm")), size = 0.5, linetype = 1, color = "black")+
  geom_text(data = soils_cca_arrows_mn, aes(x = 2*CCA1, y = 2*CCA2, label = 'Mn', fontface = "bold"), inherit.aes = FALSE)+
  geom_segment(data = soils_cca_arrows_mn, aes(x = 0, xend = 1.742239*CCA1, y = 0, yend = 1.742239 * CCA2), arrow = arrow(length = unit(0.25, "cm")), size = 0.5, linetype = 1, color = "black")+
  scale_fill_manual(values = colorRampPalette(brewer.pal(7, 'Accent'))(soils_cca_color))+
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        panel.border=element_rect(colour="black", size=1, fill=NA))+
  theme(axis.title.x=element_text(size=14, face="bold"))+
  scale_x_continuous(breaks=c(-3,-1.5,0,1.5,3),limits=c(-3,3))+
  scale_y_continuous(breaks=c(-3,-1.5,0,1.5,3),limits=c(-3,3))+
  theme(axis.title.y=element_text(size=14, face="bold"))+
  theme(axis.text.x=element_text(size=12, face="bold"))+
  theme(axis.text.y=element_text(size=12, face="bold"))
  
soils_cca_chart

tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Results/16S_soils_cca.tiff", width = 8, height = 6, units = "in", res = 300)
soils_cca_chart
dev.off()
```

#The purpose of this code chunk is to prepare alpha diversity and richness boxplots without plot information

```{r Box plots without plot}
#The purpose of this code chunk is to make the shannon diversity box plots in the concise form. We will start working with dataframe soils_alpha_diversity
soils_16s_alpha_diversity <- soils_alpha_diversity
colnames(soils_16s_alpha_diversity)[1] <- 'values'

#Create metadata
soils_treatment <- c("Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Control", "Control","Control","Control","Control","Control","Control","Control","Control","Control","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Burned","Control", "Control","Control","Control","Control","Control","Control","Control","Control","Control")
year_permanova_soils
soils_16s_alpha_without_plot <- data.frame(soils_16s_alpha_diversity, Treatment = soils_treatment, Year = year_permanova_soils)
soils_16s_alpha_without_plot$Treatment <- as.character(soils_16s_alpha_without_plot$Treatment)
soils_16s_alpha_without_plot$Treatment <- factor(soils_16s_alpha_without_plot$Treatment, levels = c("Control", "Burned"))

soils_16s_alpha_graph_without_plot <- ggplot(data = soils_16s_alpha_without_plot, aes(x = Treatment, y = values)) + geom_boxplot(aes(fill = Year), outlier.shape = NA) +
  #geom_point(position = 'jitter', aes(shape = Year))+
  ggtitle('Soils 16S')+
  xlab('Treatment')+
  ylab('Shannon Diversity Index')+
  #xlim(6.1, 7.1)+
  geom_text(data=NULL,aes(x=0.82, y=6.55, label="AB"))+
  geom_text(data=NULL,aes(x=1.18, y=6.75, label="C"))+ 
  geom_text(data=NULL,aes(x=1.82, y=6.45, label="A"))+ 
  geom_text(data=NULL,aes(x=2.18, y=7.2, label="BC"))+
  theme_linedraw()+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  theme(axis.text.x=element_text(family = 'Arial', size=12))+
  theme(axis.text.y=element_text(family = 'Arial', size=12))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))+
  scale_color_brewer(palette = 'Set1', name = 'Treatment')
soils_16s_alpha_graph_without_plot

#We have to do the same for the richness graph. We will use soils_richness_df dataframe
soils_16s_richness <- soils_richness_df

#We wanna delete the existing Treatment column
soils_16s_richness <- subset(soils_16s_richness, select = -Treatment)

#Add the same metadata previously added
soils_16s_richness_without_plot <- data.frame(soils_16s_richness, Treatment = soils_treatment, Year = year_permanova_soils)
soils_16s_richness_without_plot$Treatment <- as.character(soils_16s_richness_without_plot$Treatment)
soils_16s_richness_without_plot$Treatment <- factor(soils_16s_richness_without_plot$Treatment, levels = c("Control", "Burned"))

soils_16s_richness_graph_without_plot <- ggplot(data = soils_16s_richness_without_plot, aes(x = Treatment, y = Richness)) + geom_boxplot(aes(fill = Year), outlier.shape = NA) +
  #ggtitle('Soils 16S')+
  xlab('Treatment')+
  ylab('OTU Richness')+
  geom_text(data=NULL,aes(x=0.82, y=2400, label="A"))+
  geom_text(data=NULL,aes(x=1.18, y=3100, label="B"))+ 
  geom_text(data=NULL,aes(x=1.82, y=2400, label="A"))+ 
  geom_text(data=NULL,aes(x=2.18, y=3550, label="B"))+
  theme_linedraw()+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  scale_y_continuous(position = "right")+
  theme(axis.text.x=element_text(family = 'Arial', size=12))+
  theme(axis.text.y=element_text(family = 'Arial', size=12))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))+
  scale_color_brewer(palette = 'Set1', name = 'Treatment')

soils_16s_richness_graph_without_plot
```

#The purpose of the following code is to perform the indicator species analysis on soils data

```{r Indicator Species Analysis}
#We will use data frame rare_otu_soils_no_singletons and make a new vector to use as site classification
treatment_soils_indicator <- c("Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Burned 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Burned 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019")

#We are pretty much ready to run the analysis on soils data
library(indicspecies)
set.seed(786)
soils_indval <- multipatt(rare_otu_soils_no_singletons, treatment_soils_indicator, func = "IndVal.g", duleg = TRUE, control = how(nperm = 10000))

sink(file = "/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Tables_from_R/soils_16s_indicator_species.txt")
summary(soils_indval, indvalcomp = TRUE)
sink()

library(readxl)
soils_indicators <- read_excel("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Tables_from_R/soils_16s_indicator.xlsx", col_names = TRUE)

soils_indicators <- as.data.frame(soils_indicators)
rownames(soils_indicators) <- soils_indicators[, 2]

soils_indicators_test <- soils_indicators[, -2]

#Subset the dataframes based on treatment
soils_indic_control_2018 <- subset(soils_indicators_test, soils_indicators_test$Treatment == 'Control 2018')
soils_indic_control_2019 <- subset(soils_indicators_test, soils_indicators_test$Treatment == 'Control 2019')
soils_indic_burned_2018 <- subset(soils_indicators_test, soils_indicators_test$Treatment == 'Burned 2018')
soils_indic_burned_2019 <- subset(soils_indicators_test, soils_indicators_test$Treatment == 'Burned 2019')

#Keep first ten rows
soils_indic_control_2018_imp <- soils_indic_control_2018[1:10, ]
soils_indic_control_2019_imp <- soils_indic_control_2019[1:10, ]
soils_indic_burned_2018_imp <- soils_indic_burned_2018[1:10, ]
soils_indic_burned_2019_imp <- soils_indic_burned_2019[1:10, ]

soils_indic_imp <- rbind(soils_indic_control_2018_imp, soils_indic_control_2019_imp, soils_indic_burned_2018_imp, soils_indic_burned_2019_imp)

#Create one more dataframe only with OTU and treatment info
soils_indic_treatment <- as.data.frame(soils_indic_imp[, 1])
colnames(soils_indic_treatment) <- "Treatment"
rownames(soils_indic_treatment) <- rownames(soils_indic_imp)

soils_indic_treatment$Treatment <- factor(soils_indic_treatment$Treatment, levels = c("Control 2018", "Control 2019", "Burned 2018", "Burned 2019"))

library(RColorBrewer)
library(pheatmap)
soils_ind_heatmap <- pheatmap(soils_indic_imp[, 2:4], 
                              treeheight_row = 0, 
                              treeheight_col = 0,
                              annotation_row = soils_indic_treatment,
                              color = colorRampPalette(brewer.pal(8, "Blues"))(100),
                              cluster_rows = F, 
                              gaps_row = c(10, 20, 30))
soils_ind_heatmap

tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Results/16S_soils_indicator_species.tiff", width = 10, height = 8, units = "in", res = 300)
soils_ind_heatmap
dev.off()

#To find out the identity of the indicator species we have to merge indicator species dataframe to soils taxonomy dataframe
library(tidyr)
soils_indic_taxonomy <- merge(soils_indic_imp, soils_otu_taxonomy, by = 0)

#Subset by treatment so that we get OTUs which belong to "Burned 2019"
soils_indic_tax_burned_2019 <- soils_indic_taxonomy[soils_indic_taxonomy$Treatment == "Burned 2019", ]
soils_indic_tax_burned_2019 <- soils_indic_tax_burned_2019[order( - soils_indic_tax_burned_2019$Indicator_value), ]
```

```{r}
soils_16s_alpha_without_plot$Year <- factor(soils_16s_alpha_without_plot$Year, levels = c("2018", "2019"))
soils_alpha_x_years <- ggplot(data = soils_16s_alpha_without_plot, aes(x = Year, y = values)) + geom_boxplot(aes(fill = Treatment), outlier.shape = NA)+
  scale_color_brewer(palette = 'Set1', name = 'Treatment', direction = -1)+
  #scale_fill_manual(breaks = c("Control", "Burned"), values = c("green", "red", "green", "red"))+
  #geom_segment(x = as.numeric(soils_16s_alpha_without_plot$Year) - 0.3, xend = as.numeric(soils_16s_alpha_without_plot$Year) + 0.3, y = 6.5, yend = 6.5)+
  ggtitle('Soils 16S')+
  xlab('Treatment')+
  ylab('Shannon Diversity Index')+
  #xlim(6.1, 7.1)+
  #geom_text(data=NULL,aes(x=0.82, y=6.55, label="AB"))+
  #geom_text(data=NULL,aes(x=1.18, y=6.75, label="C"))+ 
  #geom_text(data=NULL,aes(x=1.82, y=6.45, label="A"))+ 
  #geom_text(data=NULL,aes(x=2.18, y=7.2, label="BC"))+
  theme_linedraw()+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  theme(axis.text.x=element_text(family = 'Arial', size=12))+
  theme(axis.text.y=element_text(family = 'Arial', size=12))+
  theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  theme(legend.text = element_text(size = 14, family = 'Arial'))
  #scale_color_brewer(palette = 'Set1', name = 'Treatment', direction = -1)
soils_alpha_x_years

tiff("/Users/beantkapoor/OneDrive - University of Tennessee/Dogwood Microbiome Project/Data Analysis/16s_all/Soils_16s/Results/16S_soils_new_alpha.tiff", width = 5, height = 5, units = "in", res = 200)
soils_alpha_x_years
dev.off()
```

# This is hopefully the last way I am making my boxplots from richness and alpha diversity. Let's start from a fresh dataset

```{r}
# Use soils_16s_alpha_without_plot and make a new dataframe out of it
alpha_soils <- soils_16s_alpha_without_plot

# Prepare a new vector
alpha_soils_treatment <- c("Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Pre-burned 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Control 2018","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019","Post-burned 2019", "Control 2019", "Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019","Control 2019")

#Add the above vector to alpha_soils
alpha_soils <- data.frame(alpha_treatment = alpha_soils_treatment, alpha_soils)

#Delete Treatment and year column
alpha_soils_clean <- alpha_soils[, -c(3, 4)]

#Change the column name alpha_treatment to simply Treatment
colnames(alpha_soils_clean)[1] <- "Treatment"

#Convert the variable Treatment to factor variable
alpha_soils_clean$Treatment <- factor(alpha_soils_clean$Treatment, levels = c("Control 2018", "Control 2019", "Pre-burned 2018", "Post-burned 2019"))

#Let's make the boxplot
library(ggplot2)
p_soils <- ggplot(alpha_soils_clean, aes(x = Treatment, y = values, fill = Treatment)) +
  geom_boxplot(outlier.shape = NA)+
  theme_linedraw()+
  scale_fill_manual(values = alpha(c("lightgreen","springgreen4","bisque1","firebrick3"), 0.9))+
  ggtitle('Soil 16S')+
  #xlab('Treatment')+
  #ylab('Shannon Diversity Index')+
  theme(legend.title = element_text(face = "bold", size = 14),
        legend.text = element_text(size = 14))+
  theme(plot.title = element_text(face = "bold", size = 14))+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  theme(axis.text.x=element_blank())
  #theme(axis.text.y=element_text(family = 'Arial', size=12))
  #theme(plot.margin=unit(c(1,1,-0.5,1), "cm"))
  #theme(plot.margin = unit(c(0,0,0,0), "lines"))

p_soils

tiff("/Users/beantkapoor/Desktop/16S_soils_alpha.tiff", units = "in", width = 6, height = 6, res = 300)
p_soils
dev.off()

#Have to do the same for richness. Let's start with the dataframe soils_16s_richness and add Treatment column
soils_rich <- data.frame(soils_16s_richness, Treatment = alpha_soils_treatment)
soils_rich$Treatment <- factor(soils_rich$Treatment, levels = c("Control 2018", "Control 2019", "Pre-burned 2018", "Post-burned 2019"))

#Let's make the boxplot
library(ggplot2)
r_soils <- ggplot(soils_rich, aes(x = Treatment, y = Richness, fill = Treatment)) +
  geom_boxplot(outlier.shape = NA)+
  theme_linedraw()+
  scale_fill_manual(values = alpha(c("lightgreen","springgreen4","bisque1","firebrick3"), 0.9))+
  #ggtitle('Soil 16S')+
  #xlab('Treatment')+
  #ylab('Shannon Diversity Index')+
  theme(legend.title = element_text(face = "bold", size = 14),
        legend.text = element_text(size = 14))+
  theme(plot.title = element_text(face = "bold", size = 14))+
  theme(axis.title.x=element_blank())+
  theme(axis.title.y=element_blank())+
  theme(axis.text.x=element_blank())+
  scale_y_continuous(position = "right")
  #theme(axis.text.y=element_text(family = 'Arial', size=12))+
  #theme(legend.title = element_text(family = 'Arial', face = 'bold', size = 12))+
  #theme(legend.text = element_text(size = 14, family = 'Arial'))
  #theme(plot.margin = unit(c(0,0,0,0), "lines"))

r_soils

```

#I think, if not the best way but a better way to implement the linear mixed model to find differences in soils alpha diversity

```{r Alpha diversity - Linear mixed model}
#Let's start from the dataframe soils_alpha_diversity. First change the column name
colnames(soils_alpha_diversity)[1] <- "Shannon"

#Create a vector for treatment
trt_soils <- c("Pre-burned","Pre-burned","Pre-burned","Pre-burned","Pre-burned","Pre-burned","Pre-burned","Pre-burned","Pre-burned","Pre-burned","Pre-control","Pre-control","Pre-control","Pre-control","Pre-control","Pre-control","Pre-control","Pre-control","Pre-control","Pre-control","Post-burned","Post-burned","Post-burned","Post-burned","Post-burned","Post-burned","Post-burned","Post-burned","Post-burned","Post-burned","Post-control","Post-control","Post-control","Post-control","Post-control","Post-control","Post-control","Post-control","Post-control","Post-control")

#Add these objects to the soils_alpha_diversity dataframe
trt_soils
plot_tree_soils
year_permanova_soils
soils_alpha <- data.frame(soils_alpha_diversity, Treatment = trt_soils, Year = year_permanova_soils, Plot = plot_tree_soils)

#Change the Treatment column to factor
soils_alpha$Treatment <- factor(soils_alpha$Treatment, levels = unique(soils_alpha$Treatment))

#Change the Year to numeric
soils_alpha$Year <- factor(soils_alpha$Year)

#This code is just to check if burn treatment has any effect on response varaible
library(nlme)
library(car)
soils_alpha_model <- lme(Shannon ~ Treatment, random = ~1|Plot, data = soils_alpha, method = "REML")
Anova(soils_alpha_model)
```

#T.test by treatment

```{r}
#Let's use the soils_alpha_diversity_treatment
soils_alpha_diversity_treatment$X<-sub("18","",soils_alpha_diversity_treatment$X)

soils_alpha_diversity_treatment$X<-sub("19","",soils_alpha_diversity_treatment$X)
 
library(tidyr)
soils_alpha_diversity_wide<-pivot_wider(soils_alpha_diversity_treatment, names_from = Year, values_from=soils_alpha_diversity)
 
soils_alpha_diversity_wide$yeardiffs<-soils_alpha_diversity_wide$`2019`- soils_alpha_diversity_wide$`2018`
hist(soils_alpha_diversity_wide$yeardiffs)

t.test.year.diffs_soils <-t.test(yeardiffs ~ Treatment,soils_alpha_diversity_wide)
t.test.year.diffs_soils

#Richness
soils_richness_treatment <- data.frame(soils_rich, Treatment = soils_alpha_diversity_treatment$Treatment, Year = year_permanova_soils)

#Delete the treatment column
soils_richness_treatment <- soils_richness_treatment[, -2]
colnames(soils_richness_treatment)[2] <- "Treatment"

soils_richness_treatment$Sample <- rownames(soils_richness_treatment)
soils_richness_treatment$Sample <- sub("18", "", soils_richness_treatment$Sample)
soils_richness_treatment$Sample <- sub("19", "", soils_richness_treatment$Sample)

library(tidyr)
soils_rich_wide <- pivot_wider(soils_richness_treatment, names_from = Year, values_from = Richness)
soils_rich_wide$yeardiffs <- soils_rich_wide$`2019` - soils_rich_wide$`2018`
hist(soils_rich_wide$yeardiffs)

t_test_soils_rich <- t.test(yeardiffs ~ Treatment, soils_rich_wide)
t_test_soils_rich
```

#T-test by year

```{r}
library(tidyr)
soils_alpha_t_year <- soils_alpha_diversity_treatment[, -2]
soils_alpha_t_wide <- pivot_wider(soils_alpha_t_year, names_from = Year, values_from = soils_alpha_diversity)

t_soils_alpha_year <- t.test(soils_alpha_t_wide$`2019`, soils_alpha_t_wide$`2018`, paired = TRUE, alternative = "two.sided")
t_soils_alpha_year

#Richness
soils_richness_t <- soils_richness_treatment
soils_richness_t <- soils_richness_t[, -2]
soils_richness_wide <- pivot_wider(soils_richness_t, names_from = Year, values_from = Richness)

soils_richness_t_year <- t.test(soils_richness_wide$`2019`, soils_richness_wide$`2018`, paired = TRUE, alternative = "two.sided")
soils_richness_t_year
```

